From 26a912adc35d0a91069897cf9cf0b8b0840ea196 Mon Sep 17 00:00:00 2001
From: Masahiro Yamada <yamada.masahiro@socionext.com>
Date: Tue, 29 Mar 2016 11:03:20 +0800
Subject: [PATCH 01/74] arm, powerpc: select SYS_GENERIC_BOARD

commit 09f3ca3dd53b671b009ddc4e1dad669c280094f0
[context adjustment]

We have finished Generic Board conversion for ARM and PowerPC, i.e.
all the boards have been converted except OpenRISC, SuperH, SPARC,
which have not supported Generic Board framework yet.

Select SYS_GENERIC_BOARD in arch/Kconfig and delete all the macro
defines in include/configs/*.h.

Signed-off-by: Masahiro Yamada <yamada.masahiro@socionext.com>
Integrated-by: Zhao Qiang <qiang.zhao@nxp.com>
---
 arch/Kconfig                               | 2 ++
 arch/arm/include/asm/arch-lpc32xx/config.h | 1 -
 include/configs/B4860QDS.h                 | 1 -
 include/configs/BSC9131RDB.h               | 1 -
 include/configs/BSC9132QDS.h               | 1 -
 include/configs/C29XPCIE.h                 | 1 -
 include/configs/CPCI2DP.h                  | 1 -
 include/configs/CPCI4052.h                 | 1 -
 include/configs/MIP405.h                   | 1 -
 include/configs/MPC8308RDB.h               | 1 -
 include/configs/MPC8313ERDB.h              | 1 -
 include/configs/MPC8315ERDB.h              | 1 -
 include/configs/MPC8323ERDB.h              | 1 -
 include/configs/MPC832XEMDS.h              | 1 -
 include/configs/MPC8349EMDS.h              | 1 -
 include/configs/MPC8349ITX.h               | 1 -
 include/configs/MPC837XEMDS.h              | 1 -
 include/configs/MPC837XERDB.h              | 1 -
 include/configs/MPC8536DS.h                | 1 -
 include/configs/MPC8540ADS.h               | 1 -
 include/configs/MPC8541CDS.h               | 1 -
 include/configs/MPC8544DS.h                | 1 -
 include/configs/MPC8548CDS.h               | 1 -
 include/configs/MPC8555CDS.h               | 1 -
 include/configs/MPC8560ADS.h               | 1 -
 include/configs/MPC8568MDS.h               | 1 -
 include/configs/MPC8569MDS.h               | 1 -
 include/configs/MPC8572DS.h                | 1 -
 include/configs/MPC8610HPCD.h              | 1 -
 include/configs/MPC8641HPCN.h              | 1 -
 include/configs/P1010RDB.h                 | 1 -
 include/configs/P1022DS.h                  | 1 -
 include/configs/P1023RDB.h                 | 1 -
 include/configs/P2041RDB.h                 | 1 -
 include/configs/PATI.h                     | 1 -
 include/configs/PIP405.h                   | 1 -
 include/configs/PLU405.h                   | 1 -
 include/configs/PMC405DE.h                 | 1 -
 include/configs/PMC440.h                   | 1 -
 include/configs/T102xQDS.h                 | 1 -
 include/configs/T102xRDB.h                 | 1 -
 include/configs/T1040QDS.h                 | 1 -
 include/configs/T104xRDB.h                 | 1 -
 include/configs/T208xQDS.h                 | 1 -
 include/configs/T208xRDB.h                 | 1 -
 include/configs/T4240RDB.h                 | 1 -
 include/configs/TQM5200.h                  | 1 -
 include/configs/TQM823L.h                  | 1 -
 include/configs/TQM823M.h                  | 1 -
 include/configs/TQM834x.h                  | 1 -
 include/configs/TQM850L.h                  | 1 -
 include/configs/TQM850M.h                  | 1 -
 include/configs/TQM855L.h                  | 1 -
 include/configs/TQM855M.h                  | 1 -
 include/configs/TQM860L.h                  | 1 -
 include/configs/TQM860M.h                  | 1 -
 include/configs/TQM862L.h                  | 1 -
 include/configs/TQM862M.h                  | 1 -
 include/configs/TQM866M.h                  | 1 -
 include/configs/TQM885D.h                  | 1 -
 include/configs/UCP1020.h                  | 1 -
 include/configs/VCMA9.h                    | 1 -
 include/configs/VOM405.h                   | 1 -
 include/configs/a3m071.h                   | 1 -
 include/configs/a4m072.h                   | 1 -
 include/configs/ac14xx.h                   | 1 -
 include/configs/am3517_crane.h             | 1 -
 include/configs/am3517_evm.h               | 1 -
 include/configs/amcc-common.h              | 1 -
 include/configs/apf27.h                    | 1 -
 include/configs/aria.h                     | 1 -
 include/configs/armadillo-800eva.h         | 1 -
 include/configs/aspenite.h                 | 5 -----
 include/configs/at91-sama5_common.h        | 1 -
 include/configs/at91rm9200ek.h             | 1 -
 include/configs/at91sam9260ek.h            | 1 -
 include/configs/at91sam9261ek.h            | 1 -
 include/configs/at91sam9263ek.h            | 1 -
 include/configs/at91sam9m10g45ek.h         | 1 -
 include/configs/at91sam9n12ek.h            | 1 -
 include/configs/at91sam9rlek.h             | 1 -
 include/configs/at91sam9x5ek.h             | 1 -
 include/configs/bcm28155_ap.h              | 1 -
 include/configs/bcm_ep_board.h             | 1 -
 include/configs/bur_am335x_common.h        | 1 -
 include/configs/calimain.h                 | 1 -
 include/configs/canmb.h                    | 1 -
 include/configs/cm5200.h                   | 1 -
 include/configs/cm_t35.h                   | 1 -
 include/configs/cm_t3517.h                 | 1 -
 include/configs/cm_t54.h                   | 3 ---
 include/configs/colibri_pxa270.h           | 1 -
 include/configs/colibri_vf.h               | 1 -
 include/configs/controlcenterd.h           | 1 -
 include/configs/corenet_ds.h               | 1 -
 include/configs/corvus.h                   | 1 -
 include/configs/da850evm.h                 | 1 -
 include/configs/db-88f6820-gp.h            | 1 -
 include/configs/db-mv784mp-gp.h            | 1 -
 include/configs/digsy_mtc.h                | 1 -
 include/configs/dlvision-10g.h             | 1 -
 include/configs/dlvision.h                 | 1 -
 include/configs/dns325.h                   | 1 -
 include/configs/dockstar.h                 | 1 -
 include/configs/dreamplug.h                | 1 -
 include/configs/ea20.h                     | 1 -
 include/configs/edb93xx.h                  | 1 -
 include/configs/edminiv2.h                 | 3 ---
 include/configs/ethernut5.h                | 1 -
 include/configs/exynos-common.h            | 1 -
 include/configs/flea3.h                    | 1 -
 include/configs/gdppc440etx.h              | 1 -
 include/configs/goflexhome.h               | 1 -
 include/configs/gplugd.h                   | 5 -----
 include/configs/guruplug.h                 | 1 -
 include/configs/h2200.h                    | 1 -
 include/configs/highbank.h                 | 1 -
 include/configs/hikey.h                    | 2 --
 include/configs/hrcon.h                    | 1 -
 include/configs/ib62x0.h                   | 1 -
 include/configs/iconnect.h                 | 1 -
 include/configs/ids8313.h                  | 1 -
 include/configs/imx31_phycore.h            | 1 -
 include/configs/inka4x0.h                  | 1 -
 include/configs/integrator-common.h        | 1 -
 include/configs/intip.h                    | 1 -
 include/configs/io.h                       | 1 -
 include/configs/io64.h                     | 1 -
 include/configs/iocon.h                    | 1 -
 include/configs/ipam390.h                  | 1 -
 include/configs/ipek01.h                   | 1 -
 include/configs/jupiter.h                  | 1 -
 include/configs/km/km8309-common.h         | 1 -
 include/configs/km/km8321-common.h         | 1 -
 include/configs/km/km83xx-common.h         | 1 -
 include/configs/km/km_arm.h                | 1 -
 include/configs/km/kmp204x-common.h        | 1 -
 include/configs/km82xx.h                   | 1 -
 include/configs/km8360.h                   | 1 -
 include/configs/kzm9g.h                    | 1 -
 include/configs/lacie_kw.h                 | 5 -----
 include/configs/ls1021aqds.h               | 1 -
 include/configs/ls1021atwr.h               | 1 -
 include/configs/ls2085a_common.h           | 1 -
 include/configs/lsxl.h                     | 1 -
 include/configs/lwmon5.h                   | 1 -
 include/configs/m53evk.h                   | 1 -
 include/configs/maxbcm.h                   | 1 -
 include/configs/mcx.h                      | 1 -
 include/configs/mecp5123.h                 | 1 -
 include/configs/meesc.h                    | 1 -
 include/configs/motionpro.h                | 1 -
 include/configs/mpc5121ads.h               | 1 -
 include/configs/mpc8308_p1m.h              | 1 -
 include/configs/munices.h                  | 1 -
 include/configs/mx25pdk.h                  | 1 -
 include/configs/mx31ads.h                  | 1 -
 include/configs/mx31pdk.h                  | 1 -
 include/configs/mx35pdk.h                  | 1 -
 include/configs/mx51evk.h                  | 1 -
 include/configs/mx53ard.h                  | 1 -
 include/configs/mx53evk.h                  | 1 -
 include/configs/mx53loco.h                 | 1 -
 include/configs/mx53smd.h                  | 1 -
 include/configs/mx6_common.h               | 1 -
 include/configs/mx7_common.h               | 1 -
 include/configs/mxs.h                      | 1 -
 include/configs/nas220.h                   | 1 -
 include/configs/neo.h                      | 1 -
 include/configs/nokia_rx51.h               | 1 -
 include/configs/o2dnt-common.h             | 1 -
 include/configs/omap3_evm_common.h         | 1 -
 include/configs/omap3_zoom1.h              | 1 -
 include/configs/omapl138_lcdk.h            | 1 -
 include/configs/p1_p2_rdb_pc.h             | 1 -
 include/configs/p1_twr.h                   | 1 -
 include/configs/pcm030.h                   | 1 -
 include/configs/pcm052.h                   | 1 -
 include/configs/pdm360ng.h                 | 1 -
 include/configs/picosam9g45.h              | 1 -
 include/configs/pm9261.h                   | 1 -
 include/configs/pm9263.h                   | 1 -
 include/configs/pm9g45.h                   | 1 -
 include/configs/pogo_e02.h                 | 1 -
 include/configs/qemu-ppce500.h             | 1 -
 include/configs/rcar-gen2-common.h         | 1 -
 include/configs/rk3288_common.h            | 1 -
 include/configs/rpi-common.h               | 1 -
 include/configs/s5p_goni.h                 | 1 -
 include/configs/sbc8349.h                  | 1 -
 include/configs/sbc8548.h                  | 1 -
 include/configs/sbc8641d.h                 | 1 -
 include/configs/sheevaplug.h               | 1 -
 include/configs/siemens-am33x-common.h     | 1 -
 include/configs/smartweb.h                 | 1 -
 include/configs/smdk2410.h                 | 1 -
 include/configs/smdkc100.h                 | 1 -
 include/configs/snapper9260.h              | 1 -
 include/configs/sniper.h                   | 1 -
 include/configs/socfpga_common.h           | 1 -
 include/configs/socrates.h                 | 1 -
 include/configs/spear-common.h             | 1 -
 include/configs/stm32f429-discovery.h      | 1 -
 include/configs/stv0991.h                  | 1 -
 include/configs/sunxi-common.h             | 1 -
 include/configs/t4qds.h                    | 1 -
 include/configs/tam3517-common.h           | 1 -
 include/configs/tao3530.h                  | 1 -
 include/configs/taurus.h                   | 1 -
 include/configs/tegra-common.h             | 1 -
 include/configs/ti814x_evm.h               | 1 -
 include/configs/ti816x_evm.h               | 1 -
 include/configs/ti_armv7_common.h          | 3 ---
 include/configs/tricorder.h                | 1 -
 include/configs/ts4800.h                   | 3 ---
 include/configs/uniphier.h                 | 1 -
 include/configs/usb_a9263.h                | 1 -
 include/configs/usbarmory.h                | 1 -
 include/configs/v38b.h                     | 1 -
 include/configs/ve8313.h                   | 1 -
 include/configs/vexpress_aemv8a.h          | 3 ---
 include/configs/vexpress_common.h          | 1 -
 include/configs/vf610twr.h                 | 1 -
 include/configs/vme8349.h                  | 1 -
 include/configs/woodburn_common.h          | 1 -
 include/configs/work_92105.h               | 1 -
 include/configs/x600.h                     | 1 -
 include/configs/xilinx-ppc.h               | 1 -
 include/configs/xilinx_zynqmp.h            | 1 -
 include/configs/xpedite1000.h              | 1 -
 include/configs/xpedite517x.h              | 1 -
 include/configs/xpedite520x.h              | 1 -
 include/configs/xpedite537x.h              | 1 -
 include/configs/xpedite550x.h              | 1 -
 include/configs/zmx25.h                    | 1 -
 include/configs/zynq-common.h              | 1 -
 236 files changed, 2 insertions(+), 258 deletions(-)

diff --git a/arch/Kconfig b/arch/Kconfig
index 4f73819..375f328 100644
--- a/arch/Kconfig
+++ b/arch/Kconfig
@@ -24,6 +24,7 @@ config ARM
 	select CREATE_ARCH_SYMLINK
 	select HAVE_PRIVATE_LIBGCC if !ARM64
 	select HAVE_GENERIC_BOARD
+	select SYS_GENERIC_BOARD
 	select SUPPORT_OF_CONTROL
 
 config AVR32
@@ -71,6 +72,7 @@ config PPC
 	bool "PowerPC architecture"
 	select HAVE_PRIVATE_LIBGCC
 	select HAVE_GENERIC_BOARD
+	select SYS_GENERIC_BOARD
 	select SUPPORT_OF_CONTROL
 
 config SANDBOX
diff --git a/arch/arm/include/asm/arch-lpc32xx/config.h b/arch/arm/include/asm/arch-lpc32xx/config.h
index 2d082ef..845ba4f 100644
--- a/arch/arm/include/asm/arch-lpc32xx/config.h
+++ b/arch/arm/include/asm/arch-lpc32xx/config.h
@@ -9,7 +9,6 @@
 #ifndef _LPC32XX_CONFIG_H
 #define _LPC32XX_CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Basic CPU architecture */
 #define CONFIG_ARCH_CPU_INIT
diff --git a/include/configs/B4860QDS.h b/include/configs/B4860QDS.h
index 1e458f4..6279f15 100644
--- a/include/configs/B4860QDS.h
+++ b/include/configs/B4860QDS.h
@@ -7,7 +7,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/BSC9131RDB.h b/include/configs/BSC9131RDB.h
index a7c9277..fa15f54 100644
--- a/include/configs/BSC9131RDB.h
+++ b/include/configs/BSC9131RDB.h
@@ -11,7 +11,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #ifdef CONFIG_BSC9131RDB
diff --git a/include/configs/BSC9132QDS.h b/include/configs/BSC9132QDS.h
index 804493e..d3bceb6 100644
--- a/include/configs/BSC9132QDS.h
+++ b/include/configs/BSC9132QDS.h
@@ -11,7 +11,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #ifdef CONFIG_BSC9132QDS
diff --git a/include/configs/C29XPCIE.h b/include/configs/C29XPCIE.h
index 6c39b1e..684ca25 100644
--- a/include/configs/C29XPCIE.h
+++ b/include/configs/C29XPCIE.h
@@ -12,7 +12,6 @@
 #define __CONFIG_H
 
 #define CONFIG_PHYS_64BIT
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #ifdef CONFIG_C29XPCIE
diff --git a/include/configs/CPCI2DP.h b/include/configs/CPCI2DP.h
index 638a586..9c364af 100644
--- a/include/configs/CPCI2DP.h
+++ b/include/configs/CPCI2DP.h
@@ -20,7 +20,6 @@
 #define CONFIG_405GP		1	/* This is a PPC405 CPU		*/
 
 #define	CONFIG_SYS_TEXT_BASE	0xFFFC0000
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_BOARD_EARLY_INIT_F 1	/* call board_early_init_f()	*/
diff --git a/include/configs/CPCI4052.h b/include/configs/CPCI4052.h
index 9c25751..faa342b 100644
--- a/include/configs/CPCI4052.h
+++ b/include/configs/CPCI4052.h
@@ -23,7 +23,6 @@
 #undef  CONFIG_CPCI405_6U               /* enable this for 6U boards    */
 
 #define	CONFIG_SYS_TEXT_BASE	0xFFFC0000
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_BOARD_EARLY_INIT_F 1	/* call board_early_init_f()	*/
diff --git a/include/configs/MIP405.h b/include/configs/MIP405.h
index cb21b73..0e7d268 100644
--- a/include/configs/MIP405.h
+++ b/include/configs/MIP405.h
@@ -21,7 +21,6 @@
 
 #define	CONFIG_SYS_TEXT_BASE	0xFFF80000
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /***********************************************************
  * Note that it may also be a MIP405T board which is a subset of the
diff --git a/include/configs/MPC8308RDB.h b/include/configs/MPC8308RDB.h
index 4150d5a..551b72d 100644
--- a/include/configs/MPC8308RDB.h
+++ b/include/configs/MPC8308RDB.h
@@ -9,7 +9,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/MPC8313ERDB.h b/include/configs/MPC8313ERDB.h
index d558db7..7db67c2 100644
--- a/include/configs/MPC8313ERDB.h
+++ b/include/configs/MPC8313ERDB.h
@@ -10,7 +10,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/MPC8315ERDB.h b/include/configs/MPC8315ERDB.h
index cbe10af..eb50be6 100644
--- a/include/configs/MPC8315ERDB.h
+++ b/include/configs/MPC8315ERDB.h
@@ -9,7 +9,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_SYS_NAND_U_BOOT_SIZE  (512 << 10)
diff --git a/include/configs/MPC8323ERDB.h b/include/configs/MPC8323ERDB.h
index 7547c12..59d5e56 100644
--- a/include/configs/MPC8323ERDB.h
+++ b/include/configs/MPC8323ERDB.h
@@ -9,7 +9,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/MPC832XEMDS.h b/include/configs/MPC832XEMDS.h
index 9805df7..5ab7d5f 100644
--- a/include/configs/MPC832XEMDS.h
+++ b/include/configs/MPC832XEMDS.h
@@ -7,7 +7,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/MPC8349EMDS.h b/include/configs/MPC8349EMDS.h
index 57547a4..988a900 100644
--- a/include/configs/MPC8349EMDS.h
+++ b/include/configs/MPC8349EMDS.h
@@ -13,7 +13,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/MPC8349ITX.h b/include/configs/MPC8349ITX.h
index 834786d..091ca17 100644
--- a/include/configs/MPC8349ITX.h
+++ b/include/configs/MPC8349ITX.h
@@ -40,7 +40,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #if (CONFIG_SYS_TEXT_BASE == 0xFE000000)
diff --git a/include/configs/MPC837XEMDS.h b/include/configs/MPC837XEMDS.h
index cebd175..7d1262d 100644
--- a/include/configs/MPC837XEMDS.h
+++ b/include/configs/MPC837XEMDS.h
@@ -8,7 +8,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/MPC837XERDB.h b/include/configs/MPC837XERDB.h
index 5c6ad19..ab68e63 100644
--- a/include/configs/MPC837XERDB.h
+++ b/include/configs/MPC837XERDB.h
@@ -16,7 +16,6 @@
 #define CONFIG_MPC837x		1 /* MPC837x CPU specific */
 #define CONFIG_MPC837XERDB	1
 #define CONFIG_DISPLAY_BOARDINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define	CONFIG_SYS_TEXT_BASE	0xFE000000
 
diff --git a/include/configs/MPC8536DS.h b/include/configs/MPC8536DS.h
index 1312438..2ba8c66 100644
--- a/include/configs/MPC8536DS.h
+++ b/include/configs/MPC8536DS.h
@@ -11,7 +11,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 #include "../board/freescale/common/ics307_clk.h"
 
diff --git a/include/configs/MPC8540ADS.h b/include/configs/MPC8540ADS.h
index a42c936..81377e4 100644
--- a/include/configs/MPC8540ADS.h
+++ b/include/configs/MPC8540ADS.h
@@ -18,7 +18,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* High Level Configuration Options */
diff --git a/include/configs/MPC8541CDS.h b/include/configs/MPC8541CDS.h
index 8942aae..91f3496 100644
--- a/include/configs/MPC8541CDS.h
+++ b/include/configs/MPC8541CDS.h
@@ -13,7 +13,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* High Level Configuration Options */
diff --git a/include/configs/MPC8544DS.h b/include/configs/MPC8544DS.h
index 67fac70..a64826c 100644
--- a/include/configs/MPC8544DS.h
+++ b/include/configs/MPC8544DS.h
@@ -11,7 +11,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* High Level Configuration Options */
diff --git a/include/configs/MPC8548CDS.h b/include/configs/MPC8548CDS.h
index 842afe1..f340d58 100644
--- a/include/configs/MPC8548CDS.h
+++ b/include/configs/MPC8548CDS.h
@@ -13,7 +13,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #ifdef CONFIG_36BIT
diff --git a/include/configs/MPC8555CDS.h b/include/configs/MPC8555CDS.h
index 2e8db5a..7b6820f 100644
--- a/include/configs/MPC8555CDS.h
+++ b/include/configs/MPC8555CDS.h
@@ -13,7 +13,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* High Level Configuration Options */
diff --git a/include/configs/MPC8560ADS.h b/include/configs/MPC8560ADS.h
index fabe6bf..2c1a1b1 100644
--- a/include/configs/MPC8560ADS.h
+++ b/include/configs/MPC8560ADS.h
@@ -18,7 +18,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* High Level Configuration Options */
diff --git a/include/configs/MPC8568MDS.h b/include/configs/MPC8568MDS.h
index 7a2131b..7d7fbc6 100644
--- a/include/configs/MPC8568MDS.h
+++ b/include/configs/MPC8568MDS.h
@@ -10,7 +10,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* High Level Configuration Options */
diff --git a/include/configs/MPC8569MDS.h b/include/configs/MPC8569MDS.h
index 9e38724..6965587 100644
--- a/include/configs/MPC8569MDS.h
+++ b/include/configs/MPC8569MDS.h
@@ -10,7 +10,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* High Level Configuration Options */
diff --git a/include/configs/MPC8572DS.h b/include/configs/MPC8572DS.h
index f3334ad..00c8d33 100644
--- a/include/configs/MPC8572DS.h
+++ b/include/configs/MPC8572DS.h
@@ -11,7 +11,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #include "../board/freescale/common/ics307_clk.h"
diff --git a/include/configs/MPC8610HPCD.h b/include/configs/MPC8610HPCD.h
index 9a487e0..4ce04a8 100644
--- a/include/configs/MPC8610HPCD.h
+++ b/include/configs/MPC8610HPCD.h
@@ -13,7 +13,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* High Level Configuration Options */
diff --git a/include/configs/MPC8641HPCN.h b/include/configs/MPC8641HPCN.h
index f20ee79..6f1535e 100644
--- a/include/configs/MPC8641HPCN.h
+++ b/include/configs/MPC8641HPCN.h
@@ -16,7 +16,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* High Level Configuration Options */
diff --git a/include/configs/P1010RDB.h b/include/configs/P1010RDB.h
index 8ac7000..24dff85 100644
--- a/include/configs/P1010RDB.h
+++ b/include/configs/P1010RDB.h
@@ -14,7 +14,6 @@
 #ifdef CONFIG_36BIT
 #define CONFIG_PHYS_64BIT
 #endif
-#define	CONFIG_SYS_GENERIC_BOARD
 #define	CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_P1010
diff --git a/include/configs/P1022DS.h b/include/configs/P1022DS.h
index 06b293f..c115f11 100644
--- a/include/configs/P1022DS.h
+++ b/include/configs/P1022DS.h
@@ -11,7 +11,6 @@
 
 #include "../board/freescale/common/ics307_clk.h"
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #ifdef CONFIG_36BIT
diff --git a/include/configs/P1023RDB.h b/include/configs/P1023RDB.h
index e61db18..4d87198 100644
--- a/include/configs/P1023RDB.h
+++ b/include/configs/P1023RDB.h
@@ -10,7 +10,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #ifndef CONFIG_SYS_TEXT_BASE
diff --git a/include/configs/P2041RDB.h b/include/configs/P2041RDB.h
index 5468495..de7be8b 100644
--- a/include/configs/P2041RDB.h
+++ b/include/configs/P2041RDB.h
@@ -13,7 +13,6 @@
 
 #define CONFIG_P2041RDB
 #define CONFIG_PHYS_64BIT
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 #define CONFIG_PPC_P2041
 
diff --git a/include/configs/PATI.h b/include/configs/PATI.h
index 5d0e422..81d535f 100644
--- a/include/configs/PATI.h
+++ b/include/configs/PATI.h
@@ -21,7 +21,6 @@
 
 #define	CONFIG_SYS_TEXT_BASE	0xFFF00000
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Serial Console Configuration */
 #define	CONFIG_5xx_CONS_SCI1
diff --git a/include/configs/PIP405.h b/include/configs/PIP405.h
index c9d08e6..882aae4 100644
--- a/include/configs/PIP405.h
+++ b/include/configs/PIP405.h
@@ -21,7 +21,6 @@
 
 #define	CONFIG_SYS_TEXT_BASE	0xFFF80000
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /***********************************************************
  * Clock
diff --git a/include/configs/PLU405.h b/include/configs/PLU405.h
index 3a71ff8..16dfb13 100644
--- a/include/configs/PLU405.h
+++ b/include/configs/PLU405.h
@@ -21,7 +21,6 @@
 #define CONFIG_PLU405		1	/* ...on a PLU405 board		*/
 
 #define	CONFIG_SYS_TEXT_BASE	0xFFF80000
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_BOARD_EARLY_INIT_F 1	/* call board_early_init_f()	*/
diff --git a/include/configs/PMC405DE.h b/include/configs/PMC405DE.h
index 5712298..fdc967a 100644
--- a/include/configs/PMC405DE.h
+++ b/include/configs/PMC405DE.h
@@ -12,7 +12,6 @@
 #define CONFIG_PMC405DE		1	/* ...on a PMC405DE board	*/
 
 #define	CONFIG_SYS_TEXT_BASE	0xFFFC0000
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_BOARD_EARLY_INIT_F 1	/* call board_early_init_f()	*/
diff --git a/include/configs/PMC440.h b/include/configs/PMC440.h
index 17bd180..d6ed6f7 100644
--- a/include/configs/PMC440.h
+++ b/include/configs/PMC440.h
@@ -29,7 +29,6 @@
 #define CONFIG_SYS_TEXT_BASE	0xFFF90000
 #endif
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_SYS_CLK_FREQ	33333400
diff --git a/include/configs/T102xQDS.h b/include/configs/T102xQDS.h
index 92f51f6..e11fc50 100644
--- a/include/configs/T102xQDS.h
+++ b/include/configs/T102xQDS.h
@@ -12,7 +12,6 @@
 #define __T1024QDS_H
 
 /* High Level Configuration Options */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 #define CONFIG_BOOKE
 #define CONFIG_E500			/* BOOKE e500 family */
diff --git a/include/configs/T102xRDB.h b/include/configs/T102xRDB.h
index 324f710..71dac3d 100644
--- a/include/configs/T102xRDB.h
+++ b/include/configs/T102xRDB.h
@@ -12,7 +12,6 @@
 #define __T1024RDB_H
 
 /* High Level Configuration Options */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 #define CONFIG_BOOKE
 #define CONFIG_E500			/* BOOKE e500 family */
diff --git a/include/configs/T1040QDS.h b/include/configs/T1040QDS.h
index 1f46160..8df8265 100644
--- a/include/configs/T1040QDS.h
+++ b/include/configs/T1040QDS.h
@@ -28,7 +28,6 @@
  */
 #define CONFIG_T1040QDS
 #define CONFIG_PHYS_64BIT
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #ifdef CONFIG_RAMBOOT_PBL
diff --git a/include/configs/T104xRDB.h b/include/configs/T104xRDB.h
index 5b61b56..bf19c6b 100644
--- a/include/configs/T104xRDB.h
+++ b/include/configs/T104xRDB.h
@@ -12,7 +12,6 @@
  */
 #define CONFIG_T104xRDB
 #define CONFIG_PHYS_64BIT
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_E500			/* BOOKE e500 family */
diff --git a/include/configs/T208xQDS.h b/include/configs/T208xQDS.h
index 23ca0cf..a1497a4 100644
--- a/include/configs/T208xQDS.h
+++ b/include/configs/T208xQDS.h
@@ -11,7 +11,6 @@
 #ifndef __T208xQDS_H
 #define __T208xQDS_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 #define CONFIG_ICS307_REFCLK_HZ 25000000  /* ICS307 ref clk freq */
 #define CONFIG_MMC
diff --git a/include/configs/T208xRDB.h b/include/configs/T208xRDB.h
index ef42b88..55eb4f9 100644
--- a/include/configs/T208xRDB.h
+++ b/include/configs/T208xRDB.h
@@ -11,7 +11,6 @@
 #ifndef __T2080RDB_H
 #define __T2080RDB_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 #define CONFIG_T2080RDB
 #define CONFIG_ICS307_REFCLK_HZ 25000000  /* ICS307 ref clk freq */
diff --git a/include/configs/T4240RDB.h b/include/configs/T4240RDB.h
index d43f6b7..57824ed 100644
--- a/include/configs/T4240RDB.h
+++ b/include/configs/T4240RDB.h
@@ -12,7 +12,6 @@
 
 #define CONFIG_T4240RDB
 #define CONFIG_PHYS_64BIT
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_FSL_SATA_V2
diff --git a/include/configs/TQM5200.h b/include/configs/TQM5200.h
index 5a59b80..342fdfe 100644
--- a/include/configs/TQM5200.h
+++ b/include/configs/TQM5200.h
@@ -19,7 +19,6 @@
 #define CONFIG_MPC5200		1	/* This is an MPC5200 CPU		*/
 #define CONFIG_TQM5200		1	/* ... on TQM5200 module		*/
 #undef CONFIG_TQM5200_REV100		/*  define for revision 100 modules	*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/TQM823L.h b/include/configs/TQM823L.h
index 9b9217e..21ca908 100644
--- a/include/configs/TQM823L.h
+++ b/include/configs/TQM823L.h
@@ -19,7 +19,6 @@
 
 #define CONFIG_MPC823		1	/* This is a MPC823 CPU		*/
 #define CONFIG_TQM823L		1	/* ...on a TQM8xxL module	*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0x40000000
diff --git a/include/configs/TQM823M.h b/include/configs/TQM823M.h
index 5240e0f..a9c4ff6 100644
--- a/include/configs/TQM823M.h
+++ b/include/configs/TQM823M.h
@@ -19,7 +19,6 @@
 
 #define CONFIG_MPC823		1	/* This is a MPC823 CPU		*/
 #define CONFIG_TQM823M		1	/* ...on a TQM8xxM module	*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0x40000000
diff --git a/include/configs/TQM834x.h b/include/configs/TQM834x.h
index 4164303..0b046d1 100644
--- a/include/configs/TQM834x.h
+++ b/include/configs/TQM834x.h
@@ -12,7 +12,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/TQM850L.h b/include/configs/TQM850L.h
index edadf55..a0cb753 100644
--- a/include/configs/TQM850L.h
+++ b/include/configs/TQM850L.h
@@ -19,7 +19,6 @@
 
 #define CONFIG_MPC850		1	/* This is a MPC850 CPU		*/
 #define CONFIG_TQM850L		1	/* ...on a TQM8xxL module	*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0x40000000
diff --git a/include/configs/TQM850M.h b/include/configs/TQM850M.h
index 166bb2c..d8981d8 100644
--- a/include/configs/TQM850M.h
+++ b/include/configs/TQM850M.h
@@ -19,7 +19,6 @@
 
 #define CONFIG_MPC850		1	/* This is a MPC850 CPU		*/
 #define CONFIG_TQM850M		1	/* ...on a TQM8xxM module	*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0x40000000
diff --git a/include/configs/TQM855L.h b/include/configs/TQM855L.h
index 8b16ad2..5f5466f 100644
--- a/include/configs/TQM855L.h
+++ b/include/configs/TQM855L.h
@@ -19,7 +19,6 @@
 
 #define CONFIG_MPC855		1	/* This is a MPC855 CPU		*/
 #define CONFIG_TQM855L		1	/* ...on a TQM8xxL module	*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0x40000000
diff --git a/include/configs/TQM855M.h b/include/configs/TQM855M.h
index 8a05fa4..4c9818f 100644
--- a/include/configs/TQM855M.h
+++ b/include/configs/TQM855M.h
@@ -19,7 +19,6 @@
 
 #define CONFIG_MPC855		1	/* This is a MPC855 CPU		*/
 #define CONFIG_TQM855M		1	/* ...on a TQM8xxM module	*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0x40000000
diff --git a/include/configs/TQM860L.h b/include/configs/TQM860L.h
index bf3a25b..f867029 100644
--- a/include/configs/TQM860L.h
+++ b/include/configs/TQM860L.h
@@ -19,7 +19,6 @@
 
 #define CONFIG_MPC860		1	/* This is a MPC860 CPU		*/
 #define CONFIG_TQM860L		1	/* ...on a TQM8xxL module	*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0x40000000
diff --git a/include/configs/TQM860M.h b/include/configs/TQM860M.h
index 47e5c6c..ec8f3aa 100644
--- a/include/configs/TQM860M.h
+++ b/include/configs/TQM860M.h
@@ -19,7 +19,6 @@
 
 #define CONFIG_MPC860		1	/* This is a MPC860 CPU		*/
 #define CONFIG_TQM860M		1	/* ...on a TQM8xxM module	*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0x40000000
diff --git a/include/configs/TQM862L.h b/include/configs/TQM862L.h
index fa892a9..7f39d82 100644
--- a/include/configs/TQM862L.h
+++ b/include/configs/TQM862L.h
@@ -20,7 +20,6 @@
 #define CONFIG_MPC860		1
 #define CONFIG_MPC860T		1
 #define CONFIG_MPC862		1
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_TQM862L		1	/* ...on a TQM8xxL module	*/
diff --git a/include/configs/TQM862M.h b/include/configs/TQM862M.h
index 13319f2..36ccd97 100644
--- a/include/configs/TQM862M.h
+++ b/include/configs/TQM862M.h
@@ -20,7 +20,6 @@
 #define CONFIG_MPC860		1
 #define CONFIG_MPC860T		1
 #define CONFIG_MPC862		1
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_TQM862M		1	/* ...on a TQM8xxM module	*/
diff --git a/include/configs/TQM866M.h b/include/configs/TQM866M.h
index 0e378f2..d322ba1 100644
--- a/include/configs/TQM866M.h
+++ b/include/configs/TQM866M.h
@@ -19,7 +19,6 @@
 
 #define CONFIG_MPC866		1	/* This is a MPC866 CPU		*/
 #define CONFIG_TQM866M		1	/* ...on a TQM8xxM module	*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0x40000000
diff --git a/include/configs/TQM885D.h b/include/configs/TQM885D.h
index 76ddef5..023736e 100644
--- a/include/configs/TQM885D.h
+++ b/include/configs/TQM885D.h
@@ -22,7 +22,6 @@
 
 #define CONFIG_MPC885		1	/* This is a MPC885 CPU		*/
 #define CONFIG_TQM885D		1	/* ...on a TQM88D module	*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0x40000000
diff --git a/include/configs/UCP1020.h b/include/configs/UCP1020.h
index 629be99..cf67036 100644
--- a/include/configs/UCP1020.h
+++ b/include/configs/UCP1020.h
@@ -14,7 +14,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_FSL_ELBC
diff --git a/include/configs/VCMA9.h b/include/configs/VCMA9.h
index 5b2e092..6ce88f6 100644
--- a/include/configs/VCMA9.h
+++ b/include/configs/VCMA9.h
@@ -27,7 +27,6 @@
 
 #define CONFIG_SYS_TEXT_BASE	0x0
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_ARM_CACHE_WRITETHROUGH
 
diff --git a/include/configs/VOM405.h b/include/configs/VOM405.h
index ddd6377..4d3d2ce 100644
--- a/include/configs/VOM405.h
+++ b/include/configs/VOM405.h
@@ -19,7 +19,6 @@
 #define CONFIG_VOM405		1	/* ...on a VOM405 board		*/
 
 #define	CONFIG_SYS_TEXT_BASE	0xFFFC8000
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_BOARD_EARLY_INIT_F 1	/* call board_early_init_f()	*/
diff --git a/include/configs/a3m071.h b/include/configs/a3m071.h
index fc4595b..4839ede 100644
--- a/include/configs/a3m071.h
+++ b/include/configs/a3m071.h
@@ -15,7 +15,6 @@
 #define CONFIG_MPC5200
 #define CONFIG_A3M071			/* A3M071 board */
 #define CONFIG_DISPLAY_BOARDINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define	CONFIG_SYS_TEXT_BASE	0x01000000	/* boot low for 32 MiB boards */
 
diff --git a/include/configs/a4m072.h b/include/configs/a4m072.h
index 1be78fd..0ff5164 100644
--- a/include/configs/a4m072.h
+++ b/include/configs/a4m072.h
@@ -20,7 +20,6 @@
 #define CONFIG_A4M072		1	/* ... on A4M072 board */
 #define CONFIG_MPC5200_DDR	1	/* ... use DDR RAM */
 #define CONFIG_DISPLAY_BOARDINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_TEXT_BASE	0xFE000000
 
diff --git a/include/configs/ac14xx.h b/include/configs/ac14xx.h
index f05d9ec..320be8f 100644
--- a/include/configs/ac14xx.h
+++ b/include/configs/ac14xx.h
@@ -14,7 +14,6 @@
 
 #define CONFIG_AC14XX 1
 #define CONFIG_DISPLAY_BOARDINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Memory map for the ifm AC14xx board:
diff --git a/include/configs/am3517_crane.h b/include/configs/am3517_crane.h
index 654b778..0bf0af7 100644
--- a/include/configs/am3517_crane.h
+++ b/include/configs/am3517_crane.h
@@ -19,7 +19,6 @@
 #define CONFIG_OMAP		1	/* in a TI OMAP core */
 #define CONFIG_OMAP3_AM3517CRANE	1	/* working with CRANEBOARD */
 #define CONFIG_OMAP_COMMON
-#define CONFIG_SYS_GENERIC_BOARD
 /* Common ARM Erratas */
 #define CONFIG_ARM_ERRATA_454179
 #define CONFIG_ARM_ERRATA_430973
diff --git a/include/configs/am3517_evm.h b/include/configs/am3517_evm.h
index 7049091..9aa14f4 100644
--- a/include/configs/am3517_evm.h
+++ b/include/configs/am3517_evm.h
@@ -19,7 +19,6 @@
 #define CONFIG_OMAP		1	/* in a TI OMAP core */
 #define CONFIG_OMAP3_AM3517EVM	1	/* working with AM3517EVM */
 #define CONFIG_OMAP_COMMON
-#define CONFIG_SYS_GENERIC_BOARD
 /* Common ARM Erratas */
 #define CONFIG_ARM_ERRATA_454179
 #define CONFIG_ARM_ERRATA_430973
diff --git a/include/configs/amcc-common.h b/include/configs/amcc-common.h
index 37dac7d..df4eba4 100644
--- a/include/configs/amcc-common.h
+++ b/include/configs/amcc-common.h
@@ -10,7 +10,6 @@
 #ifndef __AMCC_COMMON_H
 #define __AMCC_COMMON_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_SDRAM_BASE		0x00000000	/* _must_ be 0		*/
 #define CONFIG_SYS_MONITOR_BASE	CONFIG_SYS_TEXT_BASE	/* Start of U-Boot	*/
diff --git a/include/configs/apf27.h b/include/configs/apf27.h
index 49fbcac..5b286d2 100644
--- a/include/configs/apf27.h
+++ b/include/configs/apf27.h
@@ -20,7 +20,6 @@
  */
 #define CONFIG_MX27			/* This is a Freescale i.MX27 Chip */
 #define CONFIG_MACH_TYPE	1698	/* APF27 */
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Enable the call to miscellaneous platform dependent initialization.
diff --git a/include/configs/aria.h b/include/configs/aria.h
index 2265f3e..c38bf3c 100644
--- a/include/configs/aria.h
+++ b/include/configs/aria.h
@@ -14,7 +14,6 @@
 
 #define CONFIG_ARIA 1
 #define CONFIG_DISPLAY_BOARDINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Memory map for the ARIA board:
diff --git a/include/configs/armadillo-800eva.h b/include/configs/armadillo-800eva.h
index 7f5cbcc..dfbd60d 100644
--- a/include/configs/armadillo-800eva.h
+++ b/include/configs/armadillo-800eva.h
@@ -41,7 +41,6 @@
 #define CONFIG_USE_ARCH_MEMCPY
 #define CONFIG_TMU_TIMER
 #define CONFIG_SYS_DCACHE_OFF
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* STACK */
 #define CONFIG_SYS_INIT_SP_ADDR		0xE8083000
diff --git a/include/configs/aspenite.h b/include/configs/aspenite.h
index 62c5374..fcb5f81 100644
--- a/include/configs/aspenite.h
+++ b/include/configs/aspenite.h
@@ -11,11 +11,6 @@
 #define __CONFIG_ASPENITE_H
 
 /*
- * Generic board support
- */
-#define CONFIG_SYS_GENERIC_BOARD
-
-/*
  * Version number information
  */
 #define CONFIG_IDENT_STRING	"\nMarvell-Aspenite DB"
diff --git a/include/configs/at91-sama5_common.h b/include/configs/at91-sama5_common.h
index a5990ce..f938534 100644
--- a/include/configs/at91-sama5_common.h
+++ b/include/configs/at91-sama5_common.h
@@ -31,7 +31,6 @@
 #define CONFIG_CMD_BOOTZ
 #define CONFIG_OF_LIBFDT		/* Device Tree support */
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* general purpose I/O */
 #define CONFIG_AT91_GPIO
diff --git a/include/configs/at91rm9200ek.h b/include/configs/at91rm9200ek.h
index 7e026ef..9274024 100644
--- a/include/configs/at91rm9200ek.h
+++ b/include/configs/at91rm9200ek.h
@@ -61,7 +61,6 @@
 #define CONFIG_CMD_BOOTZ
 #define CONFIG_OF_LIBFDT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Memory Configuration
diff --git a/include/configs/at91sam9260ek.h b/include/configs/at91sam9260ek.h
index 619f5da..cc3e69c 100644
--- a/include/configs/at91sam9260ek.h
+++ b/include/configs/at91sam9260ek.h
@@ -48,7 +48,6 @@
 #define CONFIG_CMD_BOOTZ
 #define CONFIG_OF_LIBFDT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* general purpose I/O */
 #define CONFIG_ATMEL_LEGACY		/* required until (g)pio is fixed */
diff --git a/include/configs/at91sam9261ek.h b/include/configs/at91sam9261ek.h
index 4d260e9..64fb38b 100644
--- a/include/configs/at91sam9261ek.h
+++ b/include/configs/at91sam9261ek.h
@@ -33,7 +33,6 @@
 
 #define CONFIG_OF_LIBFDT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_ATMEL_LEGACY
 #define CONFIG_SYS_TEXT_BASE		0x21f00000
diff --git a/include/configs/at91sam9263ek.h b/include/configs/at91sam9263ek.h
index 2f251db..89ee9fe 100644
--- a/include/configs/at91sam9263ek.h
+++ b/include/configs/at91sam9263ek.h
@@ -48,7 +48,6 @@
 #define CONFIG_CMD_BOOTZ
 #define CONFIG_OF_LIBFDT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Hardware drivers
diff --git a/include/configs/at91sam9m10g45ek.h b/include/configs/at91sam9m10g45ek.h
index 70eaf1d..cc42c90 100644
--- a/include/configs/at91sam9m10g45ek.h
+++ b/include/configs/at91sam9m10g45ek.h
@@ -33,7 +33,6 @@
 #define CONFIG_CMD_BOOTZ
 #define CONFIG_OF_LIBFDT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* general purpose I/O */
 #define CONFIG_ATMEL_LEGACY		/* required until (g)pio is fixed */
diff --git a/include/configs/at91sam9n12ek.h b/include/configs/at91sam9n12ek.h
index 44f821b..8fa5ff5 100644
--- a/include/configs/at91sam9n12ek.h
+++ b/include/configs/at91sam9n12ek.h
@@ -31,7 +31,6 @@
 #define CONFIG_DISPLAY_CPUINFO
 
 #define CONFIG_OF_LIBFDT
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* general purpose I/O */
 #define CONFIG_AT91_GPIO
diff --git a/include/configs/at91sam9rlek.h b/include/configs/at91sam9rlek.h
index cba927a..a7a0502 100644
--- a/include/configs/at91sam9rlek.h
+++ b/include/configs/at91sam9rlek.h
@@ -34,7 +34,6 @@
 #define CONFIG_CMD_BOOTZ
 #define CONFIG_OF_LIBFDT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_ATMEL_LEGACY
 #define CONFIG_AT91_GPIO		1
diff --git a/include/configs/at91sam9x5ek.h b/include/configs/at91sam9x5ek.h
index 2068e0d..20a5581 100644
--- a/include/configs/at91sam9x5ek.h
+++ b/include/configs/at91sam9x5ek.h
@@ -29,7 +29,6 @@
 #define CONFIG_CMD_BOOTZ
 #define CONFIG_OF_LIBFDT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* general purpose I/O */
 #define CONFIG_ATMEL_LEGACY		/* required until (g)pio is fixed */
diff --git a/include/configs/bcm28155_ap.h b/include/configs/bcm28155_ap.h
index 7b48875..6ea6fb0 100644
--- a/include/configs/bcm28155_ap.h
+++ b/include/configs/bcm28155_ap.h
@@ -13,7 +13,6 @@
 /* CPU, chip, mach, etc */
 #define CONFIG_KONA
 #define CONFIG_SKIP_LOWLEVEL_INIT
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Memory configuration
diff --git a/include/configs/bcm_ep_board.h b/include/configs/bcm_ep_board.h
index ba26964..c3ebb4d 100644
--- a/include/configs/bcm_ep_board.h
+++ b/include/configs/bcm_ep_board.h
@@ -11,7 +11,6 @@
 
 #define CONFIG_SKIP_LOWLEVEL_INIT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Memory configuration
diff --git a/include/configs/bur_am335x_common.h b/include/configs/bur_am335x_common.h
index c696092..1d96a2d 100644
--- a/include/configs/bur_am335x_common.h
+++ b/include/configs/bur_am335x_common.h
@@ -26,7 +26,6 @@
 #define CONFIG_PREBOOT			"run brdefaultip"
 #define CONFIG_CMD_TIME
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_AM33XX
 #define CONFIG_OMAP
diff --git a/include/configs/calimain.h b/include/configs/calimain.h
index c8b15fb..a5b2f9e 100644
--- a/include/configs/calimain.h
+++ b/include/configs/calimain.h
@@ -18,7 +18,6 @@
 #define CONFIG_DRIVER_TI_EMAC
 #define MACH_TYPE_CALIMAIN	3528
 #define CONFIG_MACH_TYPE	MACH_TYPE_CALIMAIN
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * SoC Configuration
diff --git a/include/configs/canmb.h b/include/configs/canmb.h
index 36cd50b..2a6c09f 100644
--- a/include/configs/canmb.h
+++ b/include/configs/canmb.h
@@ -15,7 +15,6 @@
 
 #define CONFIG_MPC5200		1	/* This is a MPC5200 CPU */
 #define CONFIG_CANMB		1	/* ... on canmb board - we need this for FEC.C */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/cm5200.h b/include/configs/cm5200.h
index 4607d9f..69b2674 100644
--- a/include/configs/cm5200.h
+++ b/include/configs/cm5200.h
@@ -9,7 +9,6 @@
 #define __CONFIG_H
 
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 
diff --git a/include/configs/cm_t35.h b/include/configs/cm_t35.h
index e5fb474..e951b2d 100644
--- a/include/configs/cm_t35.h
+++ b/include/configs/cm_t35.h
@@ -25,7 +25,6 @@
 #define CONFIG_CMD_GPIO
 #define CONFIG_CM_T3X	/* working with CM-T35 and CM-T3730 */
 #define CONFIG_OMAP_COMMON
-#define CONFIG_SYS_GENERIC_BOARD
 /* Common ARM Erratas */
 #define CONFIG_ARM_ERRATA_454179
 #define CONFIG_ARM_ERRATA_430973
diff --git a/include/configs/cm_t3517.h b/include/configs/cm_t3517.h
index f3d1f43..8a10e9c 100644
--- a/include/configs/cm_t3517.h
+++ b/include/configs/cm_t3517.h
@@ -16,7 +16,6 @@
 #define CONFIG_OMAP	/* in a TI OMAP core */
 #define CONFIG_CM_T3517	/* working with CM-T3517 */
 #define CONFIG_OMAP_COMMON
-#define CONFIG_SYS_GENERIC_BOARD
 /* Common ARM Erratas */
 #define CONFIG_ARM_ERRATA_454179
 #define CONFIG_ARM_ERRATA_430973
diff --git a/include/configs/cm_t54.h b/include/configs/cm_t54.h
index 95a5a60..6d57cd6 100644
--- a/include/configs/cm_t54.h
+++ b/include/configs/cm_t54.h
@@ -18,9 +18,6 @@
 
 #undef CONFIG_SPL_OS_BOOT
 
-/* Enable Generic board */
-#define CONFIG_SYS_GENERIC_BOARD
-
 /* Device Tree defines */
 #define CONFIG_OF_LIBFDT
 #define CONFIG_OF_BOARD_SETUP
diff --git a/include/configs/colibri_pxa270.h b/include/configs/colibri_pxa270.h
index 4efc9c5..0fc24a0 100644
--- a/include/configs/colibri_pxa270.h
+++ b/include/configs/colibri_pxa270.h
@@ -14,7 +14,6 @@
  * High Level Board Configuration Options
  */
 #define	CONFIG_CPU_PXA27X		1	/* Marvell PXA270 CPU */
-#define CONFIG_SYS_GENERIC_BOARD
 #define	CONFIG_SYS_TEXT_BASE		0x0
 /* Avoid overwriting factory configuration block */
 #define CONFIG_BOARD_SIZE_LIMIT		0x40000
diff --git a/include/configs/colibri_vf.h b/include/configs/colibri_vf.h
index 2583155..c2c324c 100644
--- a/include/configs/colibri_vf.h
+++ b/include/configs/colibri_vf.h
@@ -19,7 +19,6 @@
 #define CONFIG_USE_ARCH_MEMCPY
 #define CONFIG_USE_ARCH_MEMSET
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_ARCH_MISC_INIT
 #define CONFIG_DISPLAY_CPUINFO
 #define CONFIG_DISPLAY_BOARDINFO
diff --git a/include/configs/controlcenterd.h b/include/configs/controlcenterd.h
index 9a1f6d0..4fb03e1 100644
--- a/include/configs/controlcenterd.h
+++ b/include/configs/controlcenterd.h
@@ -45,7 +45,6 @@
 #define CONFIG_CONTROLCENTERD
 #define CONFIG_MP			/* support multiple processors */
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_NO_FLASH
 #define CONFIG_ENABLE_36BIT_PHYS
diff --git a/include/configs/corenet_ds.h b/include/configs/corenet_ds.h
index 2919220..6124d04 100644
--- a/include/configs/corenet_ds.h
+++ b/include/configs/corenet_ds.h
@@ -10,7 +10,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #include "../board/freescale/common/ics307_clk.h"
diff --git a/include/configs/corvus.h b/include/configs/corvus.h
index 6805612..97797e2 100644
--- a/include/configs/corvus.h
+++ b/include/configs/corvus.h
@@ -17,7 +17,6 @@
 #include <asm/hardware.h>
 #include <linux/sizes.h>
 
-#define CONFIG_SYS_GENERIC_BOARD
 /*
  * Warning: changing CONFIG_SYS_TEXT_BASE requires
  * adapting the initial boot program.
diff --git a/include/configs/da850evm.h b/include/configs/da850evm.h
index 1cd11c0..5aa643a 100644
--- a/include/configs/da850evm.h
+++ b/include/configs/da850evm.h
@@ -258,7 +258,6 @@
 /*
  * U-Boot general configuration
  */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_MISC_INIT_R
 #define CONFIG_BOARD_EARLY_INIT_F
 #define CONFIG_BOOTFILE		"uImage" /* Boot file name */
diff --git a/include/configs/db-88f6820-gp.h b/include/configs/db-88f6820-gp.h
index 1dd4182..61446e6 100644
--- a/include/configs/db-88f6820-gp.h
+++ b/include/configs/db-88f6820-gp.h
@@ -17,7 +17,6 @@
 #define CONFIG_SYS_L2_PL310
 
 #define CONFIG_SKIP_LOWLEVEL_INIT	/* disable board lowlevel_init */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO_LATE
 
 /*
diff --git a/include/configs/db-mv784mp-gp.h b/include/configs/db-mv784mp-gp.h
index f612e72..9d951da 100644
--- a/include/configs/db-mv784mp-gp.h
+++ b/include/configs/db-mv784mp-gp.h
@@ -14,7 +14,6 @@
 #define CONFIG_DB_784MP_GP		/* Board target name for DDR training */
 
 #define CONFIG_SKIP_LOWLEVEL_INIT	/* disable board lowlevel_init */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO_LATE
 
 /*
diff --git a/include/configs/digsy_mtc.h b/include/configs/digsy_mtc.h
index a7af351..1e40ce9 100644
--- a/include/configs/digsy_mtc.h
+++ b/include/configs/digsy_mtc.h
@@ -23,7 +23,6 @@
 #define CONFIG_MPC5200		1	/* This is an MPC5200 CPU */
 #define CONFIG_DIGSY_MTC	1	/* ... on InterControl digsyMTC board */
 #define CONFIG_DISPLAY_BOARDINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Valid values for CONFIG_SYS_TEXT_BASE are:
diff --git a/include/configs/dlvision-10g.h b/include/configs/dlvision-10g.h
index e6bfe58..27c7bc6 100644
--- a/include/configs/dlvision-10g.h
+++ b/include/configs/dlvision-10g.h
@@ -24,7 +24,6 @@
 #define CONFIG_BOARD_EARLY_INIT_R
 #define CONFIG_MISC_INIT_R
 #define CONFIG_LAST_STAGE_INIT
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_CLK_FREQ	33333333 /* external frequency to pll   */
 
diff --git a/include/configs/dlvision.h b/include/configs/dlvision.h
index 0299d16..bae0583 100644
--- a/include/configs/dlvision.h
+++ b/include/configs/dlvision.h
@@ -22,7 +22,6 @@
 
 #define CONFIG_BOARD_EARLY_INIT_F	/* call board_early_init_f */
 #define CONFIG_MISC_INIT_R		/* call misc_init_r */
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_CLK_FREQ	33333333 /* external frequency to pll   */
 
diff --git a/include/configs/dns325.h b/include/configs/dns325.h
index 54be415..c59a324 100644
--- a/include/configs/dns325.h
+++ b/include/configs/dns325.h
@@ -26,7 +26,6 @@
 #define CONFIG_FEROCEON_88FR131		/* CPU Core subversion */
 #define CONFIG_KW88F6281		/* SOC Name */
 #define CONFIG_SKIP_LOWLEVEL_INIT	/* disable board lowlevel_init */
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Commands configuration
diff --git a/include/configs/dockstar.h b/include/configs/dockstar.h
index 6ebd703..efe5807 100644
--- a/include/configs/dockstar.h
+++ b/include/configs/dockstar.h
@@ -12,7 +12,6 @@
 #ifndef _CONFIG_DOCKSTAR_H
 #define _CONFIG_DOCKSTAR_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Version number information
diff --git a/include/configs/dreamplug.h b/include/configs/dreamplug.h
index 133c7f4..2815ad8 100644
--- a/include/configs/dreamplug.h
+++ b/include/configs/dreamplug.h
@@ -37,7 +37,6 @@
 #define CONFIG_KW88F6281	1	/* SOC Name */
 #define CONFIG_MACH_TYPE	MACH_TYPE_DREAMPLUG
 #define CONFIG_SKIP_LOWLEVEL_INIT	/* disable board lowlevel_init */
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Add target to build it automatically upon "make" */
 #define CONFIG_BUILD_TARGET     "u-boot.kwb"
diff --git a/include/configs/ea20.h b/include/configs/ea20.h
index 2d1f1b3..b230d9c 100644
--- a/include/configs/ea20.h
+++ b/include/configs/ea20.h
@@ -24,7 +24,6 @@
 #define CONFIG_VIDEO
 #define CONFIG_PREBOOT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * SoC Configuration
diff --git a/include/configs/edb93xx.h b/include/configs/edb93xx.h
index e3e32aa..9fce1cd 100644
--- a/include/configs/edb93xx.h
+++ b/include/configs/edb93xx.h
@@ -25,7 +25,6 @@
 #error "no board defined"
 #endif
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Initial environment and monitor configuration options. */
 #define CONFIG_BOOTDELAY		2
diff --git a/include/configs/edminiv2.h b/include/configs/edminiv2.h
index 8b38d7e..7fb1894 100644
--- a/include/configs/edminiv2.h
+++ b/include/configs/edminiv2.h
@@ -12,9 +12,6 @@
 #ifndef _CONFIG_EDMINIV2_H
 #define _CONFIG_EDMINIV2_H
 
-/* general settings */
-#define CONFIG_SYS_GENERIC_BOARD
-
 /*
  * SPL
  */
diff --git a/include/configs/ethernut5.h b/include/configs/ethernut5.h
index b49b903..d6e0456 100644
--- a/include/configs/ethernut5.h
+++ b/include/configs/ethernut5.h
@@ -12,7 +12,6 @@
 
 #include <asm/hardware.h>
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* The first stage boot loader expects u-boot running at this address. */
 #define CONFIG_SYS_TEXT_BASE	0x27000000	/* 16MB available */
diff --git a/include/configs/exynos-common.h b/include/configs/exynos-common.h
index d7bf80b..852829c 100644
--- a/include/configs/exynos-common.h
+++ b/include/configs/exynos-common.h
@@ -16,7 +16,6 @@
 #include <asm/arch/cpu.h>		/* get chip and board defs */
 #include <linux/sizes.h>
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_ARCH_CPU_INIT
 #define CONFIG_DISPLAY_CPUINFO
diff --git a/include/configs/flea3.h b/include/configs/flea3.h
index f646fee..15905b9 100644
--- a/include/configs/flea3.h
+++ b/include/configs/flea3.h
@@ -22,7 +22,6 @@
 #define CONFIG_SYS_CACHELINE_SIZE	32
 
 #define CONFIG_DISPLAY_CPUINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Only in case the value is not present in mach-types.h */
 #ifndef MACH_TYPE_FLEA3
diff --git a/include/configs/gdppc440etx.h b/include/configs/gdppc440etx.h
index bfaba42..b514278 100644
--- a/include/configs/gdppc440etx.h
+++ b/include/configs/gdppc440etx.h
@@ -32,7 +32,6 @@
 
 #define CONFIG_BOARD_EARLY_INIT_F	1	/* call board_early_init_f*/
 #define CONFIG_MISC_INIT_R		1	/* call misc_init_r()     */
-#define CONFIG_SYS_GENERIC_BOARD
 
 #undef CONFIG_ZERO_BOOTDELAY_CHECK     /* ignore keypress on bootdelay==0 */
 
diff --git a/include/configs/goflexhome.h b/include/configs/goflexhome.h
index 2a42e7c..188d2e2 100644
--- a/include/configs/goflexhome.h
+++ b/include/configs/goflexhome.h
@@ -15,7 +15,6 @@
 #ifndef _CONFIG_GOFLEXHOME_H
 #define _CONFIG_GOFLEXHOME_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Version number information
diff --git a/include/configs/gplugd.h b/include/configs/gplugd.h
index 0de6ae8..f2fd2ae 100644
--- a/include/configs/gplugd.h
+++ b/include/configs/gplugd.h
@@ -26,11 +26,6 @@
 #endif
 
 /*
- * Generic board support
- */
-#define CONFIG_SYS_GENERIC_BOARD
-
-/*
  * Version number information
  */
 #define CONFIG_IDENT_STRING	"\nMarvell-gplugD"
diff --git a/include/configs/guruplug.h b/include/configs/guruplug.h
index 5f09567..675ec52 100644
--- a/include/configs/guruplug.h
+++ b/include/configs/guruplug.h
@@ -10,7 +10,6 @@
 #ifndef _CONFIG_GURUPLUG_H
 #define _CONFIG_GURUPLUG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Version number information
diff --git a/include/configs/h2200.h b/include/configs/h2200.h
index e8dfa74..6fbbe6e 100644
--- a/include/configs/h2200.h
+++ b/include/configs/h2200.h
@@ -12,7 +12,6 @@
 #define MACH_TYPE_H2200			341
 #define CONFIG_MACH_TYPE		MACH_TYPE_H2200
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_CPU_PXA25X		1
 #define CONFIG_BOARD_H2200
 
diff --git a/include/configs/highbank.h b/include/configs/highbank.h
index 7f331a6..f02575a 100644
--- a/include/configs/highbank.h
+++ b/include/configs/highbank.h
@@ -13,7 +13,6 @@
 #define CONFIG_SYS_THUMB_BUILD
 
 #define CONFIG_SYS_NO_FLASH
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_OF_BOARD_SETUP
 #define CONFIG_FIT
diff --git a/include/configs/hikey.h b/include/configs/hikey.h
index 3af0213..1a4381b 100644
--- a/include/configs/hikey.h
+++ b/include/configs/hikey.h
@@ -14,8 +14,6 @@
 
 #include <linux/sizes.h>
 
-/* We use generic board for hikey */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_POWER
 #define CONFIG_POWER_HI6553
 
diff --git a/include/configs/hrcon.h b/include/configs/hrcon.h
index 558edfc..3cb279a 100644
--- a/include/configs/hrcon.h
+++ b/include/configs/hrcon.h
@@ -22,7 +22,6 @@
 
 #define CONFIG_IDENT_STRING	" hrcon 0.01"
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_BOARD_EARLY_INIT_F
 #define CONFIG_BOARD_EARLY_INIT_R
diff --git a/include/configs/ib62x0.h b/include/configs/ib62x0.h
index 6afe340..594fceb 100644
--- a/include/configs/ib62x0.h
+++ b/include/configs/ib62x0.h
@@ -9,7 +9,6 @@
 #ifndef _CONFIG_IB62x0_H
 #define _CONFIG_IB62x0_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Version number information
diff --git a/include/configs/iconnect.h b/include/configs/iconnect.h
index bce97dc..adf3703 100644
--- a/include/configs/iconnect.h
+++ b/include/configs/iconnect.h
@@ -9,7 +9,6 @@
 #ifndef _CONFIG_ICONNECT_H
 #define _CONFIG_ICONNECT_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Version number information
diff --git a/include/configs/ids8313.h b/include/configs/ids8313.h
index e60e753..ecb7f10 100644
--- a/include/configs/ids8313.h
+++ b/include/configs/ids8313.h
@@ -19,7 +19,6 @@
 #define CONFIG_MPC8313
 #define CONFIG_IDS8313
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_FSL_ELBC
 
diff --git a/include/configs/imx31_phycore.h b/include/configs/imx31_phycore.h
index 3bea71b..235fdd1 100644
--- a/include/configs/imx31_phycore.h
+++ b/include/configs/imx31_phycore.h
@@ -18,7 +18,6 @@
 #define CONFIG_MX31			/* This is a mx31 */
 #define CONFIG_MX31_CLK32	32000
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_DISPLAY_CPUINFO
 #define CONFIG_DISPLAY_BOARDINFO
diff --git a/include/configs/inka4x0.h b/include/configs/inka4x0.h
index d3034cf..24da83c 100644
--- a/include/configs/inka4x0.h
+++ b/include/configs/inka4x0.h
@@ -18,7 +18,6 @@
 
 #define CONFIG_MPC5200		1	/* This is an MPC5200 CPU	*/
 #define CONFIG_INKA4X0		1	/* INKA4x0 board		*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/integrator-common.h b/include/configs/integrator-common.h
index a22efd6..639a30a 100644
--- a/include/configs/integrator-common.h
+++ b/include/configs/integrator-common.h
@@ -29,7 +29,6 @@
 #define CONFIG_SETUP_MEMORY_TAGS
 #define CONFIG_OF_LIBFDT		/* enable passing a Device Tree */
 #define CONFIG_MISC_INIT_R		/* call misc_init_r during start up */
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * There are various dependencies on the core module (CM) fitted
diff --git a/include/configs/intip.h b/include/configs/intip.h
index 18d3140..a676a20 100644
--- a/include/configs/intip.h
+++ b/include/configs/intip.h
@@ -45,7 +45,6 @@
 #define CONFIG_BOARD_EARLY_INIT_F	1	/* Call board_early_init_f */
 #define CONFIG_BOARD_EARLY_INIT_R	1	/* Call board_early_init_r */
 #define CONFIG_MISC_INIT_R		1	/* Call misc_init_r */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_BOARD_TYPES		1	/* support board types */
 #define CONFIG_FIT
 #define CFG_ALT_MEMTEST
diff --git a/include/configs/io.h b/include/configs/io.h
index f5b09b6..a70ee89 100644
--- a/include/configs/io.h
+++ b/include/configs/io.h
@@ -24,7 +24,6 @@
 #define CONFIG_BOARD_EARLY_INIT_R
 #define CONFIG_MISC_INIT_R
 #define CONFIG_LAST_STAGE_INIT
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_CLK_FREQ	33333333 /* external frequency to pll   */
 
diff --git a/include/configs/io64.h b/include/configs/io64.h
index 94ccb6b..25bd014 100644
--- a/include/configs/io64.h
+++ b/include/configs/io64.h
@@ -43,7 +43,6 @@
 #define CONFIG_BOARD_EARLY_INIT_R
 #define CONFIG_MISC_INIT_R
 #define CONFIG_LAST_STAGE_INIT
-#define CONFIG_SYS_GENERIC_BOARD
 
 #undef CONFIG_ZERO_BOOTDELAY_CHECK	/* ignore keypress on bootdelay==0 */
 
diff --git a/include/configs/iocon.h b/include/configs/iocon.h
index f7ae663..a087c5b 100644
--- a/include/configs/iocon.h
+++ b/include/configs/iocon.h
@@ -23,7 +23,6 @@
 #define CONFIG_BOARD_EARLY_INIT_F
 #define CONFIG_BOARD_EARLY_INIT_R
 #define CONFIG_LAST_STAGE_INIT
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_CLK_FREQ	33333333 /* external frequency to pll   */
 
diff --git a/include/configs/ipam390.h b/include/configs/ipam390.h
index b1cd7df..583f7b3 100644
--- a/include/configs/ipam390.h
+++ b/include/configs/ipam390.h
@@ -18,7 +18,6 @@
 /*
  * Board
  */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DRIVER_TI_EMAC
 #define CONFIG_BARIX_IPAM390
 
diff --git a/include/configs/ipek01.h b/include/configs/ipek01.h
index 3f6f8dd..637bbb3 100644
--- a/include/configs/ipek01.h
+++ b/include/configs/ipek01.h
@@ -19,7 +19,6 @@
 #define CONFIG_MPX5200		1	/* MPX5200 board */
 #define CONFIG_MPC5200_DDR	1	/* use DDR RAM */
 #define CONFIG_IPEK01           	/* Motherboard is ipek01 */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0xfc000000
diff --git a/include/configs/jupiter.h b/include/configs/jupiter.h
index 5601cb4..ba96d97 100644
--- a/include/configs/jupiter.h
+++ b/include/configs/jupiter.h
@@ -15,7 +15,6 @@
 
 #define CONFIG_MPC5200		1	/* This is an MPC5200 CPU */
 #define CONFIG_JUPITER		1	/* ... on Jupiter board */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/km/km8309-common.h b/include/configs/km/km8309-common.h
index ec133f9..5cbb348 100644
--- a/include/configs/km/km8309-common.h
+++ b/include/configs/km/km8309-common.h
@@ -10,7 +10,6 @@
 #ifndef __CONFIG_KM8309_COMMON_H
 #define __CONFIG_KM8309_COMMON_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/km/km8321-common.h b/include/configs/km/km8321-common.h
index 058b0ab..cd56380 100644
--- a/include/configs/km/km8321-common.h
+++ b/include/configs/km/km8321-common.h
@@ -23,7 +23,6 @@
 #ifndef __CONFIG_KM8321_COMMON_H
 #define __CONFIG_KM8321_COMMON_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/km/km83xx-common.h b/include/configs/km/km83xx-common.h
index 940000e..cebe210 100644
--- a/include/configs/km/km83xx-common.h
+++ b/include/configs/km/km83xx-common.h
@@ -8,7 +8,6 @@
 #ifndef __CONFIG_KM83XX_H
 #define __CONFIG_KM83XX_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* include common defines/options for all Keymile boards */
diff --git a/include/configs/km/km_arm.h b/include/configs/km/km_arm.h
index 15fca1a..a54817b 100644
--- a/include/configs/km/km_arm.h
+++ b/include/configs/km/km_arm.h
@@ -20,7 +20,6 @@
 #ifndef _CONFIG_KM_ARM_H
 #define _CONFIG_KM_ARM_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* We got removed from Linux mach-types.h */
 #define MACH_TYPE_KM_KIRKWOOD          2255
diff --git a/include/configs/km/kmp204x-common.h b/include/configs/km/kmp204x-common.h
index 7536cbd..8a3c74e 100644
--- a/include/configs/km/kmp204x-common.h
+++ b/include/configs/km/kmp204x-common.h
@@ -21,7 +21,6 @@
 
 #define CONFIG_NAND_ECC_BCH
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* common KM defines */
diff --git a/include/configs/km82xx.h b/include/configs/km82xx.h
index 12f9d42..19e89ce 100644
--- a/include/configs/km82xx.h
+++ b/include/configs/km82xx.h
@@ -29,7 +29,6 @@
 #error ("Board unsupported")
 #endif
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0xFE000000
diff --git a/include/configs/km8360.h b/include/configs/km8360.h
index 04cde46..c517e78 100644
--- a/include/configs/km8360.h
+++ b/include/configs/km8360.h
@@ -9,7 +9,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /* KMBEC FPGA (PRIO) */
diff --git a/include/configs/kzm9g.h b/include/configs/kzm9g.h
index 8a451ec..94b0f03 100644
--- a/include/configs/kzm9g.h
+++ b/include/configs/kzm9g.h
@@ -14,7 +14,6 @@
 #define CONFIG_KZM_A9_GT
 #define CONFIG_RMOBILE_BOARD_STRING	"KMC KZM-A9-GT"
 #define CONFIG_MACH_TYPE MACH_TYPE_KZM9G
-#define CONFIG_SYS_GENERIC_BOARD
 
 #include <asm/arch/rmobile.h>
 
diff --git a/include/configs/lacie_kw.h b/include/configs/lacie_kw.h
index 7f878b7..f8cae40 100644
--- a/include/configs/lacie_kw.h
+++ b/include/configs/lacie_kw.h
@@ -8,11 +8,6 @@
 #define _CONFIG_LACIE_KW_H
 
 /*
- * Generic board support
- */
-#define CONFIG_SYS_GENERIC_BOARD
-
-/*
  * Machine number definition
  */
 #if defined(CONFIG_INETSPACE_V2)
diff --git a/include/configs/ls1021aqds.h b/include/configs/ls1021aqds.h
index f73900f..4f50538 100644
--- a/include/configs/ls1021aqds.h
+++ b/include/configs/ls1021aqds.h
@@ -11,7 +11,6 @@
 
 #define CONFIG_ARMV7_PSCI
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_DISPLAY_CPUINFO
 #define CONFIG_DISPLAY_BOARDINFO
diff --git a/include/configs/ls1021atwr.h b/include/configs/ls1021atwr.h
index 02cc09c..88e936b 100644
--- a/include/configs/ls1021atwr.h
+++ b/include/configs/ls1021atwr.h
@@ -11,7 +11,6 @@
 
 #define CONFIG_ARMV7_PSCI
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_DISPLAY_CPUINFO
 #define CONFIG_DISPLAY_BOARDINFO
diff --git a/include/configs/ls2085a_common.h b/include/configs/ls2085a_common.h
index 55b909c..9c8a864 100644
--- a/include/configs/ls2085a_common.h
+++ b/include/configs/ls2085a_common.h
@@ -7,7 +7,6 @@
 #ifndef __LS2_COMMON_H
 #define __LS2_COMMON_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_REMAKE_ELF
 #define CONFIG_FSL_LSCH3
diff --git a/include/configs/lsxl.h b/include/configs/lsxl.h
index 1837258..4f28f43 100644
--- a/include/configs/lsxl.h
+++ b/include/configs/lsxl.h
@@ -8,7 +8,6 @@
 #ifndef _CONFIG_LSXL_H
 #define _CONFIG_LSXL_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Version number information
diff --git a/include/configs/lwmon5.h b/include/configs/lwmon5.h
index 26136a5..f05d4d9 100644
--- a/include/configs/lwmon5.h
+++ b/include/configs/lwmon5.h
@@ -23,7 +23,6 @@
 #define CONFIG_440EPX		1		/* Specific PPC440EPx	*/
 #define CONFIG_440		1		/* ... PPC440 family	*/
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_TEXT_BASE	0xFFF80000
 #define CONFIG_HOSTNAME		lwmon5
diff --git a/include/configs/m53evk.h b/include/configs/m53evk.h
index 8853d8f..4440563 100644
--- a/include/configs/m53evk.h
+++ b/include/configs/m53evk.h
@@ -9,7 +9,6 @@
 #define __M53EVK_CONFIG_H__
 
 #define CONFIG_MX53
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_MXC_GPIO
 
 #include <asm/arch/imx-regs.h>
diff --git a/include/configs/maxbcm.h b/include/configs/maxbcm.h
index 3530a26..bc4fc71 100644
--- a/include/configs/maxbcm.h
+++ b/include/configs/maxbcm.h
@@ -12,7 +12,6 @@
  */
 #define CONFIG_ARMADA_XP		/* SOC Family Name */
 #define CONFIG_SKIP_LOWLEVEL_INIT	/* disable board lowlevel_init */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO_LATE
 
 /*
diff --git a/include/configs/mcx.h b/include/configs/mcx.h
index 0f70c47..6240666 100644
--- a/include/configs/mcx.h
+++ b/include/configs/mcx.h
@@ -25,7 +25,6 @@
 #define CONFIG_MACH_TYPE	MACH_TYPE_MCX
 #define CONFIG_BOARD_LATE_INIT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_CACHELINE_SIZE	64
 
diff --git a/include/configs/mecp5123.h b/include/configs/mecp5123.h
index a8b4b24..6e7e3ec 100644
--- a/include/configs/mecp5123.h
+++ b/include/configs/mecp5123.h
@@ -16,7 +16,6 @@
 
 #define CONFIG_MECP5123 1
 #define CONFIG_DISPLAY_BOARDINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Memory map for the MECP5123 board:
diff --git a/include/configs/meesc.h b/include/configs/meesc.h
index ab6c910..9c53fb4 100644
--- a/include/configs/meesc.h
+++ b/include/configs/meesc.h
@@ -56,7 +56,6 @@
 #define CONFIG_DISPLAY_CPUINFO			/* display cpu info and speed */
 #define CONFIG_PREBOOT				/* enable preboot variable */
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Hardware drivers
diff --git a/include/configs/motionpro.h b/include/configs/motionpro.h
index 3b97d91..376fe19 100644
--- a/include/configs/motionpro.h
+++ b/include/configs/motionpro.h
@@ -17,7 +17,6 @@
 /* CPU and board */
 #define CONFIG_MPC5200		1	/* This is a MPC5200 CPU */
 #define CONFIG_MOTIONPRO	1	/* ... on Promess Motion-PRO board */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define CONFIG_HIGH_BATS	1	/* High BATs supported */
diff --git a/include/configs/mpc5121ads.h b/include/configs/mpc5121ads.h
index 782b29d..5f4e157 100644
--- a/include/configs/mpc5121ads.h
+++ b/include/configs/mpc5121ads.h
@@ -13,7 +13,6 @@
 
 #define CONFIG_MPC5121ADS 1
 #define CONFIG_DISPLAY_BOARDINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Memory map for the MPC5121ADS board:
diff --git a/include/configs/mpc8308_p1m.h b/include/configs/mpc8308_p1m.h
index 1b09e70..1a627fc 100644
--- a/include/configs/mpc8308_p1m.h
+++ b/include/configs/mpc8308_p1m.h
@@ -9,7 +9,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/munices.h b/include/configs/munices.h
index b1b6acd..42ac029 100644
--- a/include/configs/munices.h
+++ b/include/configs/munices.h
@@ -14,7 +14,6 @@
 #define CONFIG_MPC5200		1	/* This is an MPC5200 CPU */
 #define CONFIG_MPC5200_DDR	1	/* (with DDR-SDRAM) */
 #define CONFIG_MUNICES		1	/* ... on MUNICes board */
-#define CONFIG_SYS_GENERIC_BOARD
 
 #ifndef CONFIG_SYS_TEXT_BASE
 #define CONFIG_SYS_TEXT_BASE	0xFFF00000
diff --git a/include/configs/mx25pdk.h b/include/configs/mx25pdk.h
index 0414086..93f1cd1 100644
--- a/include/configs/mx25pdk.h
+++ b/include/configs/mx25pdk.h
@@ -14,7 +14,6 @@
 #define CONFIG_MX25
 #define CONFIG_SYS_TEXT_BASE		0x81200000
 #define CONFIG_MXC_GPIO
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_TIMER_RATE		32768
 #define CONFIG_SYS_TIMER_COUNTER	\
diff --git a/include/configs/mx31ads.h b/include/configs/mx31ads.h
index c4513d2..ae9f091 100644
--- a/include/configs/mx31ads.h
+++ b/include/configs/mx31ads.h
@@ -14,7 +14,6 @@
  /* High Level Configuration Options */
 #define CONFIG_MX31		1		/* This is a mx31 */
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_DISPLAY_CPUINFO
 #define CONFIG_DISPLAY_BOARDINFO
diff --git a/include/configs/mx31pdk.h b/include/configs/mx31pdk.h
index 72d28c9..863b032 100644
--- a/include/configs/mx31pdk.h
+++ b/include/configs/mx31pdk.h
@@ -19,7 +19,6 @@
 /* High Level Configuration Options */
 #define CONFIG_MX31			/* This is a mx31 */
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_DISPLAY_CPUINFO
 #define CONFIG_DISPLAY_BOARDINFO
diff --git a/include/configs/mx35pdk.h b/include/configs/mx35pdk.h
index 6bfdaa6..6c157ab 100644
--- a/include/configs/mx35pdk.h
+++ b/include/configs/mx35pdk.h
@@ -19,7 +19,6 @@
 #define CONFIG_MX35
 
 #define CONFIG_DISPLAY_CPUINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Set TEXT at the beginning of the NOR flash */
 #define CONFIG_SYS_TEXT_BASE	0xA0000000
diff --git a/include/configs/mx51evk.h b/include/configs/mx51evk.h
index 2203c15..3cbc53e 100644
--- a/include/configs/mx51evk.h
+++ b/include/configs/mx51evk.h
@@ -26,7 +26,6 @@
 #define CONFIG_SETUP_MEMORY_TAGS
 #define CONFIG_INITRD_TAG
 #define CONFIG_REVISION_TAG
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_OF_LIBFDT
 
diff --git a/include/configs/mx53ard.h b/include/configs/mx53ard.h
index b889c25..52ba176 100644
--- a/include/configs/mx53ard.h
+++ b/include/configs/mx53ard.h
@@ -23,7 +23,6 @@
 #define CONFIG_INITRD_TAG
 #define CONFIG_REVISION_TAG
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Size of malloc() pool */
 #define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + 2 * 1024 * 1024)
diff --git a/include/configs/mx53evk.h b/include/configs/mx53evk.h
index 4f304ed..a7282f3 100644
--- a/include/configs/mx53evk.h
+++ b/include/configs/mx53evk.h
@@ -23,7 +23,6 @@
 #define CONFIG_INITRD_TAG
 #define CONFIG_REVISION_TAG
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_OF_LIBFDT
 
diff --git a/include/configs/mx53loco.h b/include/configs/mx53loco.h
index 3a65861..9e1c706 100644
--- a/include/configs/mx53loco.h
+++ b/include/configs/mx53loco.h
@@ -22,7 +22,6 @@
 #define CONFIG_SETUP_MEMORY_TAGS
 #define CONFIG_INITRD_TAG
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Size of malloc() pool */
 #define CONFIG_SYS_MALLOC_LEN		(10 * 1024 * 1024)
diff --git a/include/configs/mx53smd.h b/include/configs/mx53smd.h
index e46f2ee..e294bff 100644
--- a/include/configs/mx53smd.h
+++ b/include/configs/mx53smd.h
@@ -23,7 +23,6 @@
 #define CONFIG_INITRD_TAG
 #define CONFIG_REVISION_TAG
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Size of malloc() pool */
 #define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + 2 * 1024 * 1024)
diff --git a/include/configs/mx6_common.h b/include/configs/mx6_common.h
index ef4cb68..4ae1673 100644
--- a/include/configs/mx6_common.h
+++ b/include/configs/mx6_common.h
@@ -45,7 +45,6 @@
 
 #define CONFIG_DISPLAY_BOARDINFO
 #define CONFIG_DISPLAY_CPUINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* ATAGs */
 #define CONFIG_CMDLINE_TAG
diff --git a/include/configs/mx7_common.h b/include/configs/mx7_common.h
index ea2be49..04468b9 100644
--- a/include/configs/mx7_common.h
+++ b/include/configs/mx7_common.h
@@ -33,7 +33,6 @@
 #define CONFIG_BOARD_LATE_INIT
 
 #define CONFIG_ROM_UNIFIED_SECTIONS
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_CPUINFO
 #define CONFIG_DISPLAY_BOARDINFO
 
diff --git a/include/configs/mxs.h b/include/configs/mxs.h
index 9d823de..63fdf9e 100644
--- a/include/configs/mxs.h
+++ b/include/configs/mxs.h
@@ -40,7 +40,6 @@
 /*
  * CPU specifics
  */
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* MXS uses FDT */
 #define CONFIG_OF_LIBFDT
diff --git a/include/configs/nas220.h b/include/configs/nas220.h
index ebf6750..e5ff7e1 100644
--- a/include/configs/nas220.h
+++ b/include/configs/nas220.h
@@ -24,7 +24,6 @@
  */
 #define CONFIG_FEROCEON_88FR131		/* #define CPU Core subversion */
 #define CONFIG_KW88F6192		/* SOC Name */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_SKIP_LOWLEVEL_INIT	/* disable board lowlevel_init */
 
 /* power-on led, regulator, sata0, sata1 */
diff --git a/include/configs/neo.h b/include/configs/neo.h
index 33cee43..0d864c4 100644
--- a/include/configs/neo.h
+++ b/include/configs/neo.h
@@ -25,7 +25,6 @@
 #define CONFIG_BOARD_EARLY_INIT_R
 #define CONFIG_MISC_INIT_R
 #define CONFIG_LAST_STAGE_INIT
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_CLK_FREQ	33333333 /* external frequency to pll   */
 
diff --git a/include/configs/nokia_rx51.h b/include/configs/nokia_rx51.h
index 915df52..52369be 100644
--- a/include/configs/nokia_rx51.h
+++ b/include/configs/nokia_rx51.h
@@ -28,7 +28,6 @@
 #define CONFIG_OMAP3_RX51		/* working with RX51 */
 #define CONFIG_SYS_L2CACHE_OFF		/* pretend there is no L2 CACHE */
 #define CONFIG_OMAP_COMMON
-#define CONFIG_SYS_GENERIC_BOARD
 /* Common ARM Erratas */
 #define CONFIG_ARM_ERRATA_454179
 #define CONFIG_ARM_ERRATA_430973
diff --git a/include/configs/o2dnt-common.h b/include/configs/o2dnt-common.h
index 435f1a2..8b0dbdc 100644
--- a/include/configs/o2dnt-common.h
+++ b/include/configs/o2dnt-common.h
@@ -18,7 +18,6 @@
  */
 #define CONFIG_MPC5200
 #define CONFIG_DISPLAY_BOARDINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_MPC5XXX_CLKIN	33000000 /* running at 33.000000MHz */
 
diff --git a/include/configs/omap3_evm_common.h b/include/configs/omap3_evm_common.h
index 0ebe5c1..7e7f6f2 100644
--- a/include/configs/omap3_evm_common.h
+++ b/include/configs/omap3_evm_common.h
@@ -15,7 +15,6 @@
 #define CONFIG_OMAP			/* This is TI OMAP core */
 #define CONFIG_OMAP_GPIO
 #define CONFIG_OMAP_COMMON
-#define CONFIG_SYS_GENERIC_BOARD
 /* Common ARM Erratas */
 #define CONFIG_ARM_ERRATA_454179
 #define CONFIG_ARM_ERRATA_430973
diff --git a/include/configs/omap3_zoom1.h b/include/configs/omap3_zoom1.h
index cef2243..a59fd42 100644
--- a/include/configs/omap3_zoom1.h
+++ b/include/configs/omap3_zoom1.h
@@ -17,7 +17,6 @@
  * High Level Configuration Options
  */
 #define CONFIG_OMAP3_ZOOM1	1	/* working with Zoom MDK Rev1 */
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_NAND
 #define CONFIG_NR_DRAM_BANKS	2	/* CS1 may or may not be populated */
diff --git a/include/configs/omapl138_lcdk.h b/include/configs/omapl138_lcdk.h
index d9a3aa4..80d57f4 100644
--- a/include/configs/omapl138_lcdk.h
+++ b/include/configs/omapl138_lcdk.h
@@ -195,7 +195,6 @@
 /*
  * U-Boot general configuration
  */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_MISC_INIT_R
 #define CONFIG_BOARD_EARLY_INIT_F
 #define CONFIG_BOOTFILE		"uImage" /* Boot file name */
diff --git a/include/configs/p1_p2_rdb_pc.h b/include/configs/p1_p2_rdb_pc.h
index af89e6b..46e32e5 100644
--- a/include/configs/p1_p2_rdb_pc.h
+++ b/include/configs/p1_p2_rdb_pc.h
@@ -10,7 +10,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #ifdef CONFIG_36BIT
diff --git a/include/configs/p1_twr.h b/include/configs/p1_twr.h
index e9cc274..905ebcd 100644
--- a/include/configs/p1_twr.h
+++ b/include/configs/p1_twr.h
@@ -10,7 +10,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define	CONFIG_SYS_GENERIC_BOARD
 #define	CONFIG_DISPLAY_BOARDINFO
 #if defined(CONFIG_TWR_P1025)
 #define CONFIG_BOARDNAME "TWR-P1025"
diff --git a/include/configs/pcm030.h b/include/configs/pcm030.h
index 4ea617a..29feb7b 100644
--- a/include/configs/pcm030.h
+++ b/include/configs/pcm030.h
@@ -15,7 +15,6 @@
 #define __CONFIG_H
 
 #define CONFIG_BOARDINFO	 "phyCORE-MPC5200B-tiny"
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*-----------------------------------------------------------------------------
 High Level Configuration Options
diff --git a/include/configs/pcm052.h b/include/configs/pcm052.h
index 150698e..b32e913 100644
--- a/include/configs/pcm052.h
+++ b/include/configs/pcm052.h
@@ -13,7 +13,6 @@
 
 #define CONFIG_VF610
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_CPUINFO
 #define CONFIG_DISPLAY_BOARDINFO
 #define CONFIG_SYS_THUMB_BUILD
diff --git a/include/configs/pdm360ng.h b/include/configs/pdm360ng.h
index e7413c9..a6e7e9c 100644
--- a/include/configs/pdm360ng.h
+++ b/include/configs/pdm360ng.h
@@ -14,7 +14,6 @@
 
 #define CONFIG_PDM360NG 1
 #define CONFIG_DISPLAY_BOARDINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Memory map for the PDM360NG board:
diff --git a/include/configs/picosam9g45.h b/include/configs/picosam9g45.h
index 9e9b1d0..47613de 100644
--- a/include/configs/picosam9g45.h
+++ b/include/configs/picosam9g45.h
@@ -36,7 +36,6 @@
 #define CONFIG_CMD_BOOTZ
 #define CONFIG_OF_LIBFDT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* general purpose I/O */
 #define CONFIG_ATMEL_LEGACY		/* required until (g)pio is fixed */
diff --git a/include/configs/pm9261.h b/include/configs/pm9261.h
index f4f96a7..407b83d 100644
--- a/include/configs/pm9261.h
+++ b/include/configs/pm9261.h
@@ -20,7 +20,6 @@
 #include <asm/hardware.h>
 /* ARM asynchronous clock */
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_DISPLAY_BOARDINFO
 
diff --git a/include/configs/pm9263.h b/include/configs/pm9263.h
index bd43577..f6886a6 100644
--- a/include/configs/pm9263.h
+++ b/include/configs/pm9263.h
@@ -18,7 +18,6 @@
  */
 #include <asm/hardware.h>
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* ARM asynchronous clock */
 #define CONFIG_DISPLAY_CPUINFO
diff --git a/include/configs/pm9g45.h b/include/configs/pm9g45.h
index 09f0744..ede0ee1 100644
--- a/include/configs/pm9g45.h
+++ b/include/configs/pm9g45.h
@@ -22,7 +22,6 @@
  */
 #include <asm/hardware.h>
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_PM9G45		1	/* It's an Ronetix PM9G45 */
 #define CONFIG_SYS_AT91_CPU_NAME	"AT91SAM9G45"
diff --git a/include/configs/pogo_e02.h b/include/configs/pogo_e02.h
index 8cf3ad8..b1a3dc7 100644
--- a/include/configs/pogo_e02.h
+++ b/include/configs/pogo_e02.h
@@ -13,7 +13,6 @@
 #ifndef _CONFIG_POGO_E02_H
 #define _CONFIG_POGO_E02_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Machine type definition and ID
diff --git a/include/configs/qemu-ppce500.h b/include/configs/qemu-ppce500.h
index 45f5e78..4d1fb5e 100644
--- a/include/configs/qemu-ppce500.h
+++ b/include/configs/qemu-ppce500.h
@@ -19,7 +19,6 @@
 
 #undef CONFIG_SYS_TEXT_BASE
 #define CONFIG_SYS_TEXT_BASE	0xf01000 /* 15 MB */
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_MPC85XX_NO_RESETVEC
 
diff --git a/include/configs/rcar-gen2-common.h b/include/configs/rcar-gen2-common.h
index 7dd9260..f0a3a18 100644
--- a/include/configs/rcar-gen2-common.h
+++ b/include/configs/rcar-gen2-common.h
@@ -26,7 +26,6 @@
 #define CONFIG_CMD_EXT4_WRITE
 
 #define CONFIG_SYS_THUMB_BUILD
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Support File sytems */
 #define CONFIG_FAT_WRITE
diff --git a/include/configs/rk3288_common.h b/include/configs/rk3288_common.h
index e8aec28..36408b9 100644
--- a/include/configs/rk3288_common.h
+++ b/include/configs/rk3288_common.h
@@ -13,7 +13,6 @@
 #define CONFIG_NR_DRAM_BANKS		1
 #define CONFIG_ENV_IS_NOWHERE
 #define CONFIG_ENV_SIZE			0x2000
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_SYS_MAXARGS		16
 #define CONFIG_BAUDRATE			115200
 #define CONFIG_SYS_MALLOC_LEN		(32 << 20)
diff --git a/include/configs/rpi-common.h b/include/configs/rpi-common.h
index 06836cd..1a126a0 100644
--- a/include/configs/rpi-common.h
+++ b/include/configs/rpi-common.h
@@ -11,7 +11,6 @@
 #include <asm/arch/timer.h>
 
 /* Architecture, CPU, etc.*/
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_BCM2835
 #define CONFIG_ARCH_CPU_INIT
 
diff --git a/include/configs/s5p_goni.h b/include/configs/s5p_goni.h
index aff264b..2877e19 100644
--- a/include/configs/s5p_goni.h
+++ b/include/configs/s5p_goni.h
@@ -278,6 +278,5 @@
 
 #define CONFIG_OF_LIBFDT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #endif	/* __CONFIG_H */
diff --git a/include/configs/sbc8349.h b/include/configs/sbc8349.h
index 9d89b2a..c8b14e9 100644
--- a/include/configs/sbc8349.h
+++ b/include/configs/sbc8349.h
@@ -15,7 +15,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/sbc8548.h b/include/configs/sbc8548.h
index 0717156..3852c96 100644
--- a/include/configs/sbc8548.h
+++ b/include/configs/sbc8548.h
@@ -13,7 +13,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Top level Makefile configuration choices
diff --git a/include/configs/sbc8641d.h b/include/configs/sbc8641d.h
index eed0f49..0b22ce0 100644
--- a/include/configs/sbc8641d.h
+++ b/include/configs/sbc8641d.h
@@ -20,7 +20,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* High Level Configuration Options */
 #define CONFIG_MPC8641		1	/* MPC8641 specific */
diff --git a/include/configs/sheevaplug.h b/include/configs/sheevaplug.h
index 272e3ca..76cdbe0 100644
--- a/include/configs/sheevaplug.h
+++ b/include/configs/sheevaplug.h
@@ -10,7 +10,6 @@
 #ifndef _CONFIG_SHEEVAPLUG_H
 #define _CONFIG_SHEEVAPLUG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
  * Version number information
diff --git a/include/configs/siemens-am33x-common.h b/include/configs/siemens-am33x-common.h
index bcc8dcb..6c374eb 100644
--- a/include/configs/siemens-am33x-common.h
+++ b/include/configs/siemens-am33x-common.h
@@ -45,7 +45,6 @@
 #define CONFIG_CMD_CACHE
 #define CONFIG_CMD_TIME
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_ENV_VARS_UBOOT_CONFIG
 #ifndef CONFIG_SPL_BUILD
diff --git a/include/configs/smartweb.h b/include/configs/smartweb.h
index e11c016..44d1d5a 100644
--- a/include/configs/smartweb.h
+++ b/include/configs/smartweb.h
@@ -51,7 +51,6 @@
 
 /* setting board specific options */
 # define CONFIG_MACH_TYPE		MACH_TYPE_SMARTWEB
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_CMDLINE_EDITING
 #define CONFIG_AUTO_COMPLETE
 
diff --git a/include/configs/smdk2410.h b/include/configs/smdk2410.h
index a6bfa76..c1a45d0 100644
--- a/include/configs/smdk2410.h
+++ b/include/configs/smdk2410.h
@@ -23,7 +23,6 @@
 
 #define CONFIG_SYS_TEXT_BASE	0x0
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_ARM_CACHE_WRITETHROUGH
 
diff --git a/include/configs/smdkc100.h b/include/configs/smdkc100.h
index 4ec9c65..fc69274 100644
--- a/include/configs/smdkc100.h
+++ b/include/configs/smdkc100.h
@@ -215,6 +215,5 @@
 
 #define CONFIG_OF_LIBFDT
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #endif	/* __CONFIG_H */
diff --git a/include/configs/snapper9260.h b/include/configs/snapper9260.h
index 7758b0f..afcd16a 100644
--- a/include/configs/snapper9260.h
+++ b/include/configs/snapper9260.h
@@ -20,7 +20,6 @@
 /* ARM asynchronous clock */
 #define CONFIG_SYS_AT91_MAIN_CLOCK	18432000 /* External Crystal, in Hz */
 #define CONFIG_SYS_AT91_SLOW_CLOCK	32768
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* CPU */
 #define CONFIG_ARCH_CPU_INIT
diff --git a/include/configs/sniper.h b/include/configs/sniper.h
index a0ee5bd..f168e8f 100644
--- a/include/configs/sniper.h
+++ b/include/configs/sniper.h
@@ -34,7 +34,6 @@
  * Board
  */
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_MISC_INIT_R
 
 /*
diff --git a/include/configs/socfpga_common.h b/include/configs/socfpga_common.h
index cece095..3374683 100644
--- a/include/configs/socfpga_common.h
+++ b/include/configs/socfpga_common.h
@@ -6,7 +6,6 @@
 #ifndef __CONFIG_SOCFPGA_CYCLONE5_COMMON_H__
 #define __CONFIG_SOCFPGA_CYCLONE5_COMMON_H__
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Virtual target or real hardware */
 #undef CONFIG_SOCFPGA_VIRTUAL_TARGET
diff --git a/include/configs/socrates.h b/include/configs/socrates.h
index 50911ae..019cf30 100644
--- a/include/configs/socrates.h
+++ b/include/configs/socrates.h
@@ -27,7 +27,6 @@
 #define CONFIG_E500		1	/* BOOKE e500 family		*/
 #define CONFIG_MPC8544		1
 #define CONFIG_SOCRATES		1
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0xfff80000
diff --git a/include/configs/spear-common.h b/include/configs/spear-common.h
index edae831..1d69477 100644
--- a/include/configs/spear-common.h
+++ b/include/configs/spear-common.h
@@ -11,7 +11,6 @@
  * Common configurations used for both spear3xx as well as spear6xx
  */
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* U-boot Load Address */
 #define CONFIG_SYS_TEXT_BASE			0x00700000
diff --git a/include/configs/stm32f429-discovery.h b/include/configs/stm32f429-discovery.h
index 19d9cf9..1796995 100644
--- a/include/configs/stm32f429-discovery.h
+++ b/include/configs/stm32f429-discovery.h
@@ -10,7 +10,6 @@
 
 #define CONFIG_STM32F4
 #define CONFIG_STM32F4DISCOVERY
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_OF_LIBFDT
 
diff --git a/include/configs/stv0991.h b/include/configs/stv0991.h
index b1b6919..1526d13 100644
--- a/include/configs/stv0991.h
+++ b/include/configs/stv0991.h
@@ -13,7 +13,6 @@
 
 #define CONFIG_SYS_CORTEX_R4
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_SYS_NO_FLASH
 
 /* ram memory-related information */
diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index e1ba791..0f15321 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -154,7 +154,6 @@
 #define CONFIG_SYS_CBSIZE	1024	/* Console I/O Buffer Size */
 #define CONFIG_SYS_PBSIZE	1024	/* Print Buffer Size */
 #define CONFIG_SYS_MAXARGS	16	/* max number of command args */
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Boot Argument Buffer Size */
 #define CONFIG_SYS_BARGSIZE		CONFIG_SYS_CBSIZE
diff --git a/include/configs/t4qds.h b/include/configs/t4qds.h
index d3138fe..77b7bfb 100644
--- a/include/configs/t4qds.h
+++ b/include/configs/t4qds.h
@@ -10,7 +10,6 @@
 #ifndef __T4QDS_H
 #define __T4QDS_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 #define CONFIG_CMD_REGINFO
 
diff --git a/include/configs/tam3517-common.h b/include/configs/tam3517-common.h
index 801220a..f993433 100644
--- a/include/configs/tam3517-common.h
+++ b/include/configs/tam3517-common.h
@@ -16,7 +16,6 @@
 #define CONFIG_OMAP		/* in a TI OMAP core */
 #define CONFIG_OMAP_GPIO
 #define CONFIG_OMAP_COMMON
-#define CONFIG_SYS_GENERIC_BOARD
 /* Common ARM Erratas */
 #define CONFIG_ARM_ERRATA_454179
 #define CONFIG_ARM_ERRATA_430973
diff --git a/include/configs/tao3530.h b/include/configs/tao3530.h
index bd1c07c..97fe796 100644
--- a/include/configs/tao3530.h
+++ b/include/configs/tao3530.h
@@ -20,7 +20,6 @@
 
 #define CONFIG_OMAP_GPIO
 #define CONFIG_OMAP_COMMON
-#define CONFIG_SYS_GENERIC_BOARD
 /* Common ARM Erratas */
 #define CONFIG_ARM_ERRATA_454179
 #define CONFIG_ARM_ERRATA_430973
diff --git a/include/configs/taurus.h b/include/configs/taurus.h
index 12994c8..a5e7090 100644
--- a/include/configs/taurus.h
+++ b/include/configs/taurus.h
@@ -22,7 +22,6 @@
 #include <asm/hardware.h>
 #include <linux/sizes.h>
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #if defined(CONFIG_SPL_BUILD)
 #define CONFIG_SYS_THUMB_BUILD
diff --git a/include/configs/tegra-common.h b/include/configs/tegra-common.h
index 1c469d0..a005e6a 100644
--- a/include/configs/tegra-common.h
+++ b/include/configs/tegra-common.h
@@ -130,7 +130,6 @@
 #define CONFIG_SPL_SERIAL_SUPPORT
 #define CONFIG_SPL_GPIO_SUPPORT
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_BOARD_EARLY_INIT_F
 #define CONFIG_BOARD_LATE_INIT
 
diff --git a/include/configs/ti814x_evm.h b/include/configs/ti814x_evm.h
index ceeb7fe..7fa35a1 100644
--- a/include/configs/ti814x_evm.h
+++ b/include/configs/ti814x_evm.h
@@ -21,7 +21,6 @@
 #define CONFIG_SYS_NO_FLASH
 #define CONFIG_OMAP
 #define CONFIG_OMAP_COMMON
-#define CONFIG_SYS_GENERIC_BOARD
 
 #include <asm/arch/omap.h>
 
diff --git a/include/configs/ti816x_evm.h b/include/configs/ti816x_evm.h
index 4c19289..01d8233 100644
--- a/include/configs/ti816x_evm.h
+++ b/include/configs/ti816x_evm.h
@@ -15,7 +15,6 @@
 #define CONFIG_SYS_NO_FLASH
 #define CONFIG_OMAP
 #define CONFIG_OMAP_COMMON
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_ARCH_CPU_INIT
 
diff --git a/include/configs/ti_armv7_common.h b/include/configs/ti_armv7_common.h
index 6dc6515..ae1d9d2 100644
--- a/include/configs/ti_armv7_common.h
+++ b/include/configs/ti_armv7_common.h
@@ -17,9 +17,6 @@
 #ifndef __CONFIG_TI_ARMV7_COMMON_H__
 #define __CONFIG_TI_ARMV7_COMMON_H__
 
-/* Common define for many platforms. */
-#define CONFIG_SYS_GENERIC_BOARD
-
 /*
  * We typically do not contain NOR flash.  In the cases where we do, we
  * undefine this later.
diff --git a/include/configs/tricorder.h b/include/configs/tricorder.h
index f04b572..02b0e63 100644
--- a/include/configs/tricorder.h
+++ b/include/configs/tricorder.h
@@ -38,7 +38,6 @@
 #include <asm/arch/cpu.h>		/* get chip and board defs */
 #include <asm/arch/omap.h>
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Display CPU and Board information */
 #define CONFIG_DISPLAY_CPUINFO
diff --git a/include/configs/ts4800.h b/include/configs/ts4800.h
index 21f1555..fbca8a6 100644
--- a/include/configs/ts4800.h
+++ b/include/configs/ts4800.h
@@ -37,9 +37,6 @@
 #define CONFIG_INITRD_TAG
 #define CONFIG_REVISION_TAG
 
-/* use common/board_f.c instead of arch/<arch>/lib/<board>.c */
-#define CONFIG_SYS_GENERIC_BOARD
-
 /*
  * Size of malloc() pool
  */
diff --git a/include/configs/uniphier.h b/include/configs/uniphier.h
index 9109b7f..1e71703 100644
--- a/include/configs/uniphier.h
+++ b/include/configs/uniphier.h
@@ -82,7 +82,6 @@
 /* serial console configuration */
 #define CONFIG_BAUDRATE			115200
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #if !defined(CONFIG_SPL_BUILD)
 #define CONFIG_USE_ARCH_MEMSET
diff --git a/include/configs/usb_a9263.h b/include/configs/usb_a9263.h
index 10a1664..54daede 100644
--- a/include/configs/usb_a9263.h
+++ b/include/configs/usb_a9263.h
@@ -34,7 +34,6 @@
 #define CONFIG_DISPLAY_CPUINFO
 
 #define CONFIG_OF_LIBFDT
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_SYS_TEXT_BASE		0x23f00000
 
 /*
diff --git a/include/configs/usbarmory.h b/include/configs/usbarmory.h
index 714e3e2..4af9f36 100644
--- a/include/configs/usbarmory.h
+++ b/include/configs/usbarmory.h
@@ -16,7 +16,6 @@
 #define CONFIG_DISPLAY_BOARDINFO
 #define CONFIG_BOARD_EARLY_INIT_F
 #define CONFIG_OF_LIBFDT
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_MXC_GPIO
 
 #include <asm/arch/imx-regs.h>
diff --git a/include/configs/v38b.h b/include/configs/v38b.h
index 3e51b9f..f154208 100644
--- a/include/configs/v38b.h
+++ b/include/configs/v38b.h
@@ -14,7 +14,6 @@
  */
 #define CONFIG_MPC5200			1	/* This is an MPC5200 CPU */
 #define CONFIG_V38B			1	/* ...on V38B board */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE		0xFF000000
diff --git a/include/configs/ve8313.h b/include/configs/ve8313.h
index a88216c..f5df4fb 100644
--- a/include/configs/ve8313.h
+++ b/include/configs/ve8313.h
@@ -13,7 +13,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/vexpress_aemv8a.h b/include/configs/vexpress_aemv8a.h
index 0f2f1a3..a99d712 100644
--- a/include/configs/vexpress_aemv8a.h
+++ b/include/configs/vexpress_aemv8a.h
@@ -8,9 +8,6 @@
 #ifndef __VEXPRESS_AEMV8A_H
 #define __VEXPRESS_AEMV8A_H
 
-/* We use generic board and device manager for v8 Versatile Express */
-#define CONFIG_SYS_GENERIC_BOARD
-
 #ifdef CONFIG_TARGET_VEXPRESS64_BASE_FVP
 #ifndef CONFIG_SEMIHOSTING
 #error CONFIG_TARGET_VEXPRESS64_BASE_FVP requires CONFIG_SEMIHOSTING
diff --git a/include/configs/vexpress_common.h b/include/configs/vexpress_common.h
index 98f6ae9..cec510c 100644
--- a/include/configs/vexpress_common.h
+++ b/include/configs/vexpress_common.h
@@ -122,7 +122,6 @@
 #define CONFIG_SETUP_MEMORY_TAGS	1
 #define CONFIG_SYS_L2CACHE_OFF		1
 #define CONFIG_INITRD_TAG		1
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Size of malloc() pool */
 #define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + 128 * 1024)
diff --git a/include/configs/vf610twr.h b/include/configs/vf610twr.h
index 7f4260a..df570d9 100644
--- a/include/configs/vf610twr.h
+++ b/include/configs/vf610twr.h
@@ -13,7 +13,6 @@
 
 #define CONFIG_VF610
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_CPUINFO
 #define CONFIG_DISPLAY_BOARDINFO
 
diff --git a/include/configs/vme8349.h b/include/configs/vme8349.h
index bc4a998..d53cf42 100644
--- a/include/configs/vme8349.h
+++ b/include/configs/vme8349.h
@@ -18,7 +18,6 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 /*
diff --git a/include/configs/woodburn_common.h b/include/configs/woodburn_common.h
index fc46565..c25b710 100644
--- a/include/configs/woodburn_common.h
+++ b/include/configs/woodburn_common.h
@@ -21,7 +21,6 @@
 #define CONFIG_SYS_CACHELINE_SIZE	32
 
 #define CONFIG_DISPLAY_CPUINFO
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Only in case the value is not present in mach-types.h */
 #ifndef MACH_TYPE_FLEA3
diff --git a/include/configs/work_92105.h b/include/configs/work_92105.h
index 15ee284..f77d065 100644
--- a/include/configs/work_92105.h
+++ b/include/configs/work_92105.h
@@ -103,7 +103,6 @@
 /*
  * U-Boot General Configurations
  */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_SYS_LONGHELP
 #define CONFIG_SYS_CBSIZE		1024
 #define CONFIG_SYS_PBSIZE		\
diff --git a/include/configs/x600.h b/include/configs/x600.h
index 73ba67c..8534766 100644
--- a/include/configs/x600.h
+++ b/include/configs/x600.h
@@ -16,7 +16,6 @@
  */
 #define CONFIG_SPEAR600				/* SPEAr600 SoC */
 #define CONFIG_X600				/* on X600 board */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_SYS_THUMB_BUILD
 
 #include <asm/arch/hardware.h>
diff --git a/include/configs/xilinx-ppc.h b/include/configs/xilinx-ppc.h
index 309d68d..571e329 100644
--- a/include/configs/xilinx-ppc.h
+++ b/include/configs/xilinx-ppc.h
@@ -12,7 +12,6 @@
 #ifndef __CONFIG_XLX_H
 #define __CONFIG_XLX_H
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /*
 #define DEBUG
diff --git a/include/configs/xilinx_zynqmp.h b/include/configs/xilinx_zynqmp.h
index da87188..adb072e 100644
--- a/include/configs/xilinx_zynqmp.h
+++ b/include/configs/xilinx_zynqmp.h
@@ -17,7 +17,6 @@
 
 #define CONFIG_SYS_NO_FLASH
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 /* Generic Interrupt Controller Definitions */
 #define CONFIG_GICV2
diff --git a/include/configs/xpedite1000.h b/include/configs/xpedite1000.h
index 4fafb5a..cafd01c 100644
--- a/include/configs/xpedite1000.h
+++ b/include/configs/xpedite1000.h
@@ -22,7 +22,6 @@
 #define CONFIG_440GX		1		/* 440 GX */
 #define CONFIG_BOARD_EARLY_INIT_F 1		/* Call board_pre_init	*/
 #define CONFIG_SYS_CLK_FREQ	33333333	/* external freq to pll */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0xFFF80000
diff --git a/include/configs/xpedite517x.h b/include/configs/xpedite517x.h
index b5d1126..9ba8b7a 100644
--- a/include/configs/xpedite517x.h
+++ b/include/configs/xpedite517x.h
@@ -23,7 +23,6 @@
 #define CONFIG_BAT_RW		1	/* Use common BAT rw code */
 #define CONFIG_HIGH_BATS	1	/* High BATs supported and enabled */
 #define CONFIG_ALTIVEC		1
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #define	CONFIG_SYS_TEXT_BASE	0xfff00000
diff --git a/include/configs/xpedite520x.h b/include/configs/xpedite520x.h
index 34a124c..574c961 100644
--- a/include/configs/xpedite520x.h
+++ b/include/configs/xpedite520x.h
@@ -21,7 +21,6 @@
 #define CONFIG_SYS_BOARD_NAME	"XPedite5200"
 #define CONFIG_SYS_FORM_PMC_XMC	1
 #define CONFIG_BOARD_EARLY_INIT_R	/* Call board_pre_init */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #ifndef CONFIG_SYS_TEXT_BASE
diff --git a/include/configs/xpedite537x.h b/include/configs/xpedite537x.h
index 0d24234..4899856 100644
--- a/include/configs/xpedite537x.h
+++ b/include/configs/xpedite537x.h
@@ -21,7 +21,6 @@
 #define CONFIG_SYS_BOARD_NAME	"XPedite5370"
 #define CONFIG_SYS_FORM_3U_VPX	1
 #define CONFIG_BOARD_EARLY_INIT_R	/* Call board_pre_init */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #ifndef CONFIG_SYS_TEXT_BASE
diff --git a/include/configs/xpedite550x.h b/include/configs/xpedite550x.h
index f30d199..00536e9 100644
--- a/include/configs/xpedite550x.h
+++ b/include/configs/xpedite550x.h
@@ -22,7 +22,6 @@
 #define CONFIG_SYS_FORM_PMC_XMC	1
 #define CONFIG_PRPMC_PCI_ALIAS	"pci0"	/* Processor PMC interface on pci0 */
 #define CONFIG_BOARD_EARLY_INIT_R	/* Call board_pre_init */
-#define CONFIG_SYS_GENERIC_BOARD
 #define CONFIG_DISPLAY_BOARDINFO
 
 #ifndef CONFIG_SYS_TEXT_BASE
diff --git a/include/configs/zmx25.h b/include/configs/zmx25.h
index 81b9ce4..1d4c0c9 100644
--- a/include/configs/zmx25.h
+++ b/include/configs/zmx25.h
@@ -15,7 +15,6 @@
 #define CONFIG_MX25
 #define CONFIG_SYS_TEXT_BASE		0xA0000000
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #define CONFIG_SYS_TIMER_RATE		32768
 #define CONFIG_SYS_TIMER_COUNTER	\
diff --git a/include/configs/zynq-common.h b/include/configs/zynq-common.h
index e7ab50a..d02fa77 100644
--- a/include/configs/zynq-common.h
+++ b/include/configs/zynq-common.h
@@ -376,6 +376,5 @@
 
 #define CONFIG_SYS_UBOOT_START	CONFIG_SYS_TEXT_BASE
 
-#define CONFIG_SYS_GENERIC_BOARD
 
 #endif /* __CONFIG_ZYNQ_COMMON_H */
-- 
2.1.0.27.g96db324

